diff -Naur gcstar.orig/share/applications/gcstar.desktop gcstar/share/applications/gcstar.desktop
--- gcstar.orig/share/applications/gcstar.desktop	2009-11-20 01:30:36.021699429 +0200
+++ gcstar/share/applications/gcstar.desktop	2009-11-20 01:35:30.162574380 +0200
@@ -1,10 +1,13 @@
 [Desktop Entry]
 Version=1.0
 Name=GCstar Collections Manager
+Name[tr]=GCstar
 Name[es]=Gestor de colecciones GCstar
 Name[fr]=GCstar gestionnaire de collections
 GenericName=Personal Collections Manager
+GenericName[tr]=Kişisel Kolleksiyon Yöneticisi
 Comment=Manage your collections of movies, games, books, music and more
+Comment[tr]=Film, oyun, kitap ve müzik kolleksiyonlarınızı yönetin
 Comment[en]=Manage your collections of movies, games, books, music and more
 Comment[es]=Organice sus colecciones de películas, juegos, libros, música y más
 Comment[fr]=Gérer vos collections de films, jeux vidéos, livres, musique,...
@@ -14,5 +17,5 @@
 Terminal=false
 StartupNotify=true
 Type=Application
-Categories=Office;
-MimeType=application/x-gcstar
+Categories=Office;Database;GTK;
+MimeType=application/x-gcstar; 
