diff -Naur -x '*~' gcstar/bin/gcstar gcstar-fhs/bin/gcstar
--- gcstar/bin/gcstar	2011-01-16 17:21:34.000000000 +0100
+++ gcstar-fhs/bin/gcstar	2012-08-11 13:58:16.468350307 +0200
@@ -43,14 +43,10 @@
 use FindBin qw($RealBin);
 use POSIX qw(locale_h);
 
-$ENV{GCS_BIN_DIR} = $RealBin;
-($ENV{GCS_LIB_DIR} = $ENV{GCS_BIN_DIR}) =~ s/bin\/?$/lib\/gcstar/;
-($ENV{GCS_SHARE_DIR} = $ENV{GCS_BIN_DIR}) =~ s/bin\/?$/share\/gcstar/;
-use lib File::Spec->canonpath("$RealBin/../lib/gcstar");
-if (exists $ENV{PAR_TEMP})
-{
-    unshift @INC, "$RealBin/../lib/gcstar";
-}
+$ENV{GCS_BIN_DIR} = '/usr/bin';
+$ENV{GCS_LIB_DIR} = '/usr/share/gcstar/lib';
+$ENV{GCS_SHARE_DIR} = '/usr/share/gcstar';
+use lib '/usr/share/gcstar/lib';
 
 # For win32, redirect all the output to files
 if ($^O =~ /win32/i)