--- waflib/Options.py.orig	2011-12-18 00:36:07.000000000 +0200
+++ waflib/Options.py	2012-01-08 13:16:01.699000027 +0200
@@ -157,7 +157,7 @@
 		:return: the amount of cpu cores
 		:rtype: int
 		"""
-		count = int(os.environ.get('JOBS', 0))
+		count = int(os.environ.get('JOBS', 0).replace('-j',''))
 		if count < 1:
 			if 'NUMBER_OF_PROCESSORS' in os.environ:
 				# on Windows, use the NUMBER_OF_PROCESSORS environment variable
