
 Makefile                 |   16 +++++-----------
 scripts/kconfig/Makefile |    4 ----
 2 files changed, 5 insertions(+), 15 deletions(-)

diff -Nurp linux-5.1-rc5/Makefile linux-5.1-rc5-mrp/Makefile
--- linux-5.1-rc5/Makefile	2019-04-15 15:52:25.430382341 +0300
+++ linux-5.1-rc5-mrp/Makefile	2019-04-15 16:11:29.901266267 +0300
@@ -1061,8 +1061,7 @@ include/config/kernel.release: $(srctree
 # Carefully list dependencies so we do not try to build scripts twice
 # in parallel
 PHONY += scripts
-scripts: scripts_basic scripts_dtc
-	$(Q)$(MAKE) $(build)=$(@)
+scripts:
 
 # Things we need to do before we recursively start building the kernel
 # or the modules are listed in "prepare".
@@ -1096,7 +1095,7 @@ prepare0: archprepare
 	$(Q)$(MAKE) $(build)=.
 
 # All the preparing..
-prepare: prepare0 prepare-objtool
+prepare:
 
 # Support for using generic headers in asm-generic
 asm-generic := -f $(srctree)/scripts/Makefile.asm-generic obj
@@ -1336,13 +1335,8 @@ endif # CONFIG_MODULES
 CLEAN_DIRS  += $(MODVERDIR) include/ksym
 
 # Directories & files removed with 'make mrproper'
-MRPROPER_DIRS  += include/config usr/include include/generated          \
-		  arch/*/include/generated .tmp_objdiff
-MRPROPER_FILES += .config .config.old .version \
-		  Module.symvers tags TAGS cscope* GPATH GTAGS GRTAGS GSYMS \
-		  signing_key.pem signing_key.priv signing_key.x509	\
-		  x509.genkey extra_certificates signing_key.x509.keyid	\
-		  signing_key.x509.signer vmlinux-gdb.py
+MRPROPER_DIRS  += ""
+MRPROPER_FILES += ""
 
 # clean - Delete most, but leave enough to build external modules
 #
@@ -1364,7 +1358,7 @@ clean: archclean vmlinuxclean
 #
 mrproper: rm-dirs  := $(wildcard $(MRPROPER_DIRS))
 mrproper: rm-files := $(wildcard $(MRPROPER_FILES))
-mrproper-dirs      := $(addprefix _mrproper_,scripts)
+mrproper-dirs      := $(addprefix _mrproper_)
 
 PHONY += $(mrproper-dirs) mrproper
 $(mrproper-dirs):
diff -Nurp linux-5.1-rc5/scripts/kconfig/Makefile linux-5.1-rc5-mrp/scripts/kconfig/Makefile
--- linux-5.1-rc5/scripts/kconfig/Makefile	2019-04-15 15:52:26.396402498 +0300
+++ linux-5.1-rc5-mrp/scripts/kconfig/Makefile	2019-04-15 16:11:29.901266267 +0300
@@ -20,16 +20,12 @@ endif
 unexport CONFIG_
 
 xconfig: $(obj)/qconf
-	$< $(silent) $(Kconfig)
 
 gconfig: $(obj)/gconf
-	$< $(silent) $(Kconfig)
 
 menuconfig: $(obj)/mconf
-	$< $(silent) $(Kconfig)
 
 config: $(obj)/conf
-	$< $(silent) --oldaskconfig $(Kconfig)
 
 nconfig: $(obj)/nconf
 	$< $(silent) $(Kconfig)
