--- linux-5.1-rc6/tools/testing/selftests/Makefile.orig
+++ linux-5.1-rc6/tools/testing/selftests/Makefile
@@ -33,7 +33,6 @@ TARGETS += netfilter
 TARGETS += networking/timestamping
 TARGETS += nsfs
 TARGETS += pidfd
-TARGETS += powerpc
 TARGETS += proc
 TARGETS += pstore
 TARGETS += ptrace
@@ -42,7 +41,6 @@ TARGETS += rtc
 TARGETS += seccomp
 TARGETS += sigaltstack
 TARGETS += size
-TARGETS += sparc64
 TARGETS += splice
 TARGETS += static_keys
 TARGETS += sync
