--- src/lidwatcher.cpp.old	2014-04-25 19:00:12.000000000 +0300
+++ src/lidwatcher.cpp	2014-04-28 22:09:12.630464141 +0300
@@ -27,8 +27,8 @@
 #include <QTimer>
 #include <QCoreApplication>
 #include <lxqt/lxqtautostartentry.h>
-#include <qt4/QtCore/qprocess.h>
-#include <qt4/QtCore/qtextstream.h>
+#include <QtCore/qprocess.h>
+#include <QtCore/qtextstream.h>
 #include <QDebug>
 #include "lidwatcher.h"
 #include "../config/powermanagementsettings.h"
