Index: anjuta-3.6.1/manuals/reference/libanjuta/libanjuta-docs.sgml
===================================================================
--- anjuta-3.6.1.orig/manuals/reference/libanjuta/libanjuta-docs.sgml
+++ anjuta-3.6.1/manuals/reference/libanjuta/libanjuta-docs.sgml
@@ -64,17 +64,13 @@
     <xi:include href="xml/anjuta-serializer.xml"/>
     <xi:include href="xml/anjuta-session.xml"/>
     <xi:include href="xml/anjuta-utils.xml"/>
-    <xi:include href="xml/anjuta-marshal.xml"/>
-    <xi:include href="xml/anjuta-enum-types.xml"/>
     <xi:include href="xml/resources.xml"/>
     <xi:include href="xml/anjuta-encodings.xml"/>
     <xi:include href="xml/anjuta-debug.xml"/>
-    <xi:include href="xml/anjuta-error.xml"/>
     <xi:include href="xml/anjuta-command.xml"/>
     <xi:include href="xml/anjuta-sync-command.xml"/>
     <xi:include href="xml/anjuta-async-command.xml"/>
     <xi:include href="xml/anjuta-vcs-status-tree-view.xml"/>
-    <xi:include href="xml/anjuta-message-area.xml"/>
     <xi:include href="xml/anjuta-convert.xml"/>
     <xi:include href="xml/anjuta-version.xml"/>
     <xi:include href="xml/anjuta-autogen.xml"/>
@@ -86,12 +82,10 @@
     <xi:include href="xml/anjuta-c-module.xml"/>
     <xi:include href="xml/anjuta-utils-priv.xml"/>
     <xi:include href="xml/e-splash.xml"/>
-    <xi:include href="xml/cell-renderer-captioned-image.xml"/>
   </chapter>
 
   <chapter id="plugin-interfaces">
     <title>Plugin interfaces</title>
-	<xi:include href="xml/ianjuta-bookmark.xml"/>
 	<xi:include href="xml/ianjuta-buildable.xml"/>
 	<xi:include href="xml/ianjuta-builder.xml"/>
 	<xi:include href="xml/ianjuta-debugger.xml"/>
@@ -130,7 +124,6 @@
 	<xi:include href="xml/ianjuta-iterable-tree.xml"/>
 	<xi:include href="xml/ianjuta-language.xml"/>
 	<xi:include href="xml/ianjuta-loader.xml"/>
-	<xi:include href="xml/ianjuta-macro.xml"/>
 	<xi:include href="xml/ianjuta-markable.xml"/>
 	<xi:include href="xml/ianjuta-message-manager.xml"/>
 	<xi:include href="xml/ianjuta-message-view.xml"/>
