--- metacity-2.27.0/src/core/display.c	2009-06-07 21:35:13.623787399 -0400
+++ hacked/src/core/display.c	2009-06-06 00:20:58.889278832 -0400
@@ -3307,7 +3307,7 @@
   meta_display_set_grab_op_cursor (display, screen, op, FALSE, grab_xwindow,
                                    timestamp);
 
-  if (!display->grab_have_pointer)
+  if (!display->grab_have_pointer && !grab_op_is_keyboard (op))
     {
       meta_topic (META_DEBUG_WINDOW_OPS,
                   "XGrabPointer() failed\n");