--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -197,7 +197,7 @@ libmuffininclude_HEADERS =			\
 muffin_theme_viewer_SOURCES=  \
 	ui/theme-viewer.c
 
-bin_PROGRAMS=muffin muffin-theme-viewer
+bin_PROGRAMS=muffin-theme-viewer
 
 muffin_SOURCES = core/muffin.c
 muffin_LDADD = $(MUFFIN_LIBS) libmuffin.la -lclutter-glx-1.0
@@ -257,13 +257,6 @@ testgradient_LDADD = $(MUFFIN_LIBS) libm
 testasyncgetprop_LDADD = $(MUFFIN_LIBS)
 
 
-@INTLTOOL_DESKTOP_RULE@
-
-desktopfilesdir=$(datadir)/applications
-desktopfiles_in_files=muffin.desktop.in
-desktopfiles_files=$(desktopfiles_in_files:.desktop.in=.desktop)
-desktopfiles_DATA = $(desktopfiles_files)
-
 gsettings_SCHEMAS = org.cinnamon.muffin.gschema.xml
 @INTLTOOL_XML_NOMERGE_RULE@
 @GSETTINGS_RULES@
@@ -279,7 +272,6 @@ VARIABLES=stock_maximize_data $(srcdir)/
 BUILT_SOURCES = inlinepixbufs.h
 CLEANFILES =					\
 	inlinepixbufs.h				\
-	muffin.desktop				\
 	org.cinnamon.muffin.gschema.xml		\
 	$(xml_DATA)				\
 	$(muffin_built_sources)			\
--- a/src/tools/Makefile.am
+++ b/src/tools/Makefile.am
@@ -20,7 +20,7 @@ muffin_mag_SOURCES=					\
 muffin_grayscale_SOURCES=				\
 	muffin-grayscale.c
 
-bin_PROGRAMS=muffin-message muffin-window-demo
+bin_PROGRAMS=muffin-window-demo
 
 ## cheesy hacks I use, don't really have any business existing. ;-)
 noinst_PROGRAMS=muffin-mag muffin-grayscale
--- a/doc/man/Makefile.am
+++ b/doc/man/Makefile.am
@@ -1,4 +1,4 @@
-man_MANS = muffin.1 muffin-theme-viewer.1 \
-	   muffin-window-demo.1 muffin-message.1
+man_MANS = muffin-theme-viewer.1 \
+	   muffin-window-demo.1
 
 EXTRA_DIST = $(man_MANS)