--- gpsd-3.10.orig/SConstruct	2013-11-22 14:10:01.000000000 +0100
+++ gpsd-3.10/SConstruct	2014-02-07 19:33:32.000000000 +0100
@@ -231,6 +231,9 @@ for (name, default, help) in pathopts:
 
 env['VERSION'] = gpsd_version
 env['PYTHON'] = sys.executable
+env['PLATFORM'] = "posix"
+env['SHLIBSUFFIX'] = ".so"
+env['SHLINKFLAGS'] = "-shared"
 
 # Set defaults from environment.  Note that scons doesn't cope well
 # with multi-word CPPFLAGS/LDFLAGS/SHLINKFLAGS values; you'll have to