--- lightdm.conf	2013-10-31 00:34:00.000000000 +0200
+++ /data/lightdm.conf	2013-12-25 15:23:33.900119706 +0200
@@ -20,6 +20,7 @@
 #greeter-user=lightdm
 #minimum-display-number=0
 #minimum-vt=7
+minimum-vt=1
 #lock-memory=true
 #user-authority-in-system-dir=false
 #guest-account-script=guest-account
@@ -27,8 +28,10 @@
 #run-directory=/var/run/lightdm
 #cache-directory=/var/cache/lightdm
 #sessions-directory=/usr/share/lightdm/sessions:/usr/share/xsessions
+sessions-directory=/usr/share/xsessions
 #remote-sessions-directory=/usr/share/lightdm/remote-sessions
 #greeters-directory=/usr/share/lightdm/greeters:/usr/share/xgreeters
+greeters-directory=/usr/share/xgreeters
 
 #
 # Seat defaults
@@ -86,6 +89,7 @@
 #unity-compositor-command=unity-system-compositor
 #unity-compositor-timeout=60
 #greeter-session=example-gtk-gnome
+greeter-session=lightdm-kde-greeter
 #greeter-hide-users=false
 #greeter-allow-guest=true
 #greeter-show-manual-login=false
@@ -93,7 +97,9 @@
 #user-session=default
 #allow-guest=true
 #guest-session=UNIMPLEMENTED
+pam-service=lightdm-autologin
 #session-wrapper=lightdm-session
+session-wrapper=/etc/lightdm/Xsession
 #greeter-wrapper=
 #guest-wrapper=
 #display-setup-script=