diff -Nuar anjuta-2.26.2.2.orig/src/Makefile.am anjuta-2.26.2.2/src/Makefile.am
--- anjuta-2.26.2.2.orig/src/Makefile.am	2009-07-07 17:17:01.000000000 +0300
+++ anjuta-2.26.2.2/src/Makefile.am	2009-07-07 17:20:29.000000000 +0300
@@ -50,7 +50,7 @@
 	$(GLADE_LIBS)\
 	$(top_builddir)/libanjuta/libanjuta.la
 
-anjuta_LDFLAGS = $(ANJUTA_LDFLAGS)
+anjuta_LDFLAGS = $(ANJUTA_LDFLAGS) `pkg-config --libs unique-1.0`
 
 prefs_glade_files = anjuta.glade
 include $(top_srcdir)/scripts/build-schemas.mk
