diff --git a/Source/WebCore/rendering/RenderBlockLineLayout.cpp b/Source/WebCore/rendering/RenderBlockLineLayout.cpp
index 8550ca49544f..b04b3c72c450 100644
--- a/Source/WebCore/rendering/RenderBlockLineLayout.cpp
+++ b/Source/WebCore/rendering/RenderBlockLineLayout.cpp
@@ -28,6 +28,7 @@
 #include "InlineIterator.h"
 #include "InlineTextBox.h"
 #include "Logging.h"
+#include "Font.h"
 #include "RenderArena.h"
 #include "RenderCombineText.h"
 #include "RenderCounter.h"
@@ -60,6 +61,17 @@ using namespace std;
 using namespace WTF;
 using namespace Unicode;
 
+namespace WTF {
+
+// allow compilation of OwnPtr<TextLayout> in source files that don't have access to the TextLayout class definition
+template <> void deleteOwnedPtr<WebCore::TextLayout>(WebCore::TextLayout* ptr)
+{
+    WebCore::Font::deleteLayout(ptr);
+}
+
+}
+
+
 namespace WebCore {
 
 // We don't let our line box tree for a single line get any deeper than this.
