--- a/files/usr/bin/cinnamon-launcher
+++ b/files/usr/bin/cinnamon-launcher
@@ -1,7 +1,7 @@
 #! /usr/bin/python2
 # -*- coding=utf-8 -*-
 
-FALLBACK_COMMAND = "metacity"
+FALLBACK_COMMAND = "icewm"
 FALLBACK_ARGS = ("--replace",)
 
 import os, sys, gettext
@@ -28,11 +28,6 @@ if __name__ == "__main__":
         exit_status = os.waitpid(cinnamon_pid, 0)[1]
         if exit_status != 0:
             if os.fork() == 0:
-                if os.path.exists("/usr/bin/gnome-panel"):
-                    os.system("gnome-panel --replace &")
-                elif os.path.exists("/usr/bin/tint2"):
-                    os.system("killall tint2")
-                    os.system("tint2 &")
                 os.execvp(FALLBACK_COMMAND, (FALLBACK_COMMAND,) + FALLBACK_ARGS)
             else:
                 if confirm_restart():
