--- a/files/usr/share/cinnamon/cinnamon-settings/cinnamon-settings.py
+++ b/files/usr/share/cinnamon/cinnamon-settings/cinnamon-settings.py
@@ -76,12 +76,9 @@ CONTROL_CENTER_MODULES = [
 STANDALONE_MODULES = [
 #         Label                          Executable                          Icon                Category        Keywords for filter
     [_("Printers"),                      "system-config-printer",        "cs-printer",         "hardware",       _("printers, laser, inkjet")],
-    [_("Firewall"),                      "gufw",                         "cs-firewall",        "admin",          _("firewall, block, filter, programs")],
-    [_("Languages"),                     "mintlocale",                   "cs-language",        "prefs",          _("language, install, foreign")],
-    [_("Input Method"),                  "mintlocale im",                "cs-input-method",    "prefs",          _("language, install, foreign, input, method, chinese, korean, japanese, typing")],
-    [_("Login Window"),                  "gksu /usr/sbin/mdmsetup",      "cs-login",           "admin",          _("login, mdm, gdm, manager, user, password, startup, switch")],
-    [_("Driver Manager"),                "mintdrivers",                  "cs-drivers",         "admin",          _("video, driver, wifi, card, hardware, proprietary, nvidia, radeon, nouveau, fglrx")],
-    [_("Software Sources"),              "mintsources",                  "cs-sources",         "admin",          _("ppa, repository, package, source, download")],
+    [_("Login Window") + " (LightDM)",   "xdg-su -c /usr/bin/lightdm-gtk-greeter-settings", "cs-login", "admin", _("login, lightdm, gtk-greeter, manager, user, password, startup, switch")],
+    [_("Login Window") + " (MDM)",       "xdg-su -c /usr/sbin/mdmsetup", "cs-login",           "admin",          _("login, mdm, gdm, manager, user, password, startup, switch")],
+    [_("YaST"),                          "xdg-su -c /sbin/yast2",        "yast",               "admin",          _("system, admin, yast2, settings")],
     [_("Users and Groups"),              "cinnamon-settings-users",      "cs-user-accounts",   "admin",          _("user, users, account, accounts, group, groups, password")],
     [_("Bluetooth"),                     "blueberry",                    "cs-bluetooth",       "hardware",       _("bluetooth, dongle, transfer, mobile")]
 ]
