diff -Nuar OpenCASCADE651.orig/ros/src/Draw/Draw_Window.cxx OpenCASCADE651/ros/src/Draw/Draw_Window.cxx
--- OpenCASCADE651.orig/ros/src/Draw/Draw_Window.cxx	2011-04-28 19:43:26.000000000 +0300
+++ OpenCASCADE651/ros/src/Draw/Draw_Window.cxx	2013-08-20 22:10:05.401466216 +0300
@@ -68,12 +68,13 @@
       code = Tcl_Eval(Interp, promptCmd);
       outChannel = Tcl_GetStdChannel(TCL_STDOUT);
       errChannel = Tcl_GetStdChannel(TCL_STDERR);
-      if (code != TCL_OK) {
-        if (errChannel) {
-          Tcl_Write(errChannel, Interp->result, -1);
-          Tcl_Write(errChannel, "\n", 1);
-        }
-        Tcl_AddErrorInfo(Interp,
+       if (code != TCL_OK) {
+         if (errChannel) {
+-          Tcl_Write(errChannel, Interp->result, -1);
++          Tcl_Write(errChannel, Tcl_GetStringResult(Interp), -1);
+           Tcl_Write(errChannel, "\n", 1);
+         }
+         Tcl_AddErrorInfo(Interp,
                          "\n    (script that generates prompt)");
         goto defaultPrompt;
       }
@@ -1041,13 +1042,14 @@
 
   Tcl_StaticPackage(interp, "Tk", Tk_Init, (Tcl_PackageInitProc *) NULL);
 
-  mainWindow =
-  Tk_MainWindow(interp) ;
-  if (mainWindow == NULL) {
-    fprintf(stderr, "%s\n", interp->result);
-    exit(1);
-  }
-  Tk_Name(mainWindow) =
+   mainWindow =
+   Tk_MainWindow(interp) ;
+   if (mainWindow == NULL) {
+-    fprintf(stderr, "%s\n", interp->result);
++    fprintf(stderr, "%s\n", Tcl_GetStringResult(interp));
+     exit(1);
+   }
+   Tk_Name(mainWindow) =
   Tk_GetUid(Tk_SetAppName(mainWindow,
                           "Draw")) ;
 
@@ -2063,11 +2065,14 @@
   {
     try
     {
-      OCC_CATCH_SIGNALS
-      Standard_Integer res = Tk_Init (interp);
-      if (res != TCL_OK)
-      {
-        cout << "tkLoop: error in Tk initialization. Tcl reported: " << interp->result << endl;
+     OCC_CATCH_SIGNALS
+     Standard_Integer res = Tk_Init(interp) ;
+     if (res != TCL_OK)
+-      cout << "tkLoop: error in Tk initialization. Tcl reported: " << interp->result << endl;
++      cout << "tkLoop: error in Tk initialization. Tcl reported: " << Tcl_GetStringResult(interp) << endl;
+ 
+   } catch  (Standard_Failure) {
+     cout <<"tkLoop: exception in TK_Init "<<endl;
       }
     }
     catch (Standard_Failure)
@@ -2075,12 +2080,13 @@
       cout << "tkLoop: exception in TK_Init\n";
     }
     Tcl_StaticPackage (interp, "Tk", Tk_Init, (Tcl_PackageInitProc* ) NULL);
-    mainWindow = Tk_MainWindow (interp);
-    if (mainWindow == NULL)
-    {
-      fprintf (stderr, "%s\n", interp->result);
-      cout << "tkLoop: Tk_MainWindow() returned NULL. Exiting...\n";
-      Tcl_Exit (0);
+   mainWindow =
+     Tk_MainWindow(interp) ;
+   if (mainWindow == NULL) {
+-        fprintf(stderr, "%s\n", interp->result);
++        fprintf(stderr, "%s\n", Tcl_GetStringResult(interp));
+         cout << "tkLoop: Tk_MainWindow() returned NULL. Exiting..." << endl;
+         Tcl_Exit(0);
     }
     Tk_Name(mainWindow) = Tk_GetUid (Tk_SetAppName (mainWindow, "Draw"));
   }
