--- yali/gui/Ui/setupuserswidget.ui~	2013-04-05 15:54:03.085980339 +0200
+++ yali/gui/Ui/setupuserswidget.ui	2013-04-05 15:55:21.028977491 +0200
@@ -792,21 +792,6 @@
                          <property name="currentRow">
                           <number>-1</number>
                          </property>
-                         <item>
-                          <property name="text">
-                           <string>Yeni Öge</string>
-                          </property>
-                         </item>
-                         <item>
-                          <property name="text">
-                           <string>Yeni Öge</string>
-                          </property>
-                         </item>
-                         <item>
-                          <property name="text">
-                           <string>Yeni Öge</string>
-                          </property>
-                         </item>
                         </widget>
                        </item>
                        <item>
