--- qtbase/src/xml/sax/qxml.cpp.orig	2014-05-15 20:12:16.000000000 +0300
+++ qtbase/src/xml/sax/qxml.cpp	2014-05-23 00:15:09.006626439 +0300
@@ -433,7 +433,7 @@
     // for the DTD currently being parsed.
     static const int dtdRecursionLimit = 2;
     // The maximum amount of characters an entity value may contain, after expansion.
-    static const int entityCharacterLimit = 1024;
+    static const int entityCharacterLimit = 65536;
 
     const QString &string();
     void stringClear();
