--- libshout-idjc-2.3.0/COPYING.orig	2012-12-01 11:36:09.772899434 +0200
+++ libshout-idjc-2.3.0/COPYING	2012-12-01 11:36:25.532905386 +0200
@@ -2,7 +2,7 @@
 		       Version 2, June 1991
 
  Copyright (C) 1991 Free Software Foundation, Inc.
-    		    59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+ 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA
  Everyone is permitted to copy and distribute verbatim copies
  of this license document, but changing it is not allowed.
 
