From 780556c3e1047acbcdfa588980f52c73d9aae89a Mon Sep 17 00:00:00 2001
From: Vladimir 'phcoder' Serbinenko <phcoder@gmail.com>
Date: Wed, 8 May 2013 11:01:11 +0200
Subject: [PATCH 3/5] 	* configure.ac: Don't disable extended registers on
 emu.

---
 ChangeLog    | 4 ++++
 configure.ac | 2 +-
 2 files changed, 5 insertions(+), 1 deletion(-)

diff --git a/configure.ac b/configure.ac
index 80549b0..fa0e00a 100644
--- a/configure.ac
+++ b/configure.ac
@@ -466,7 +466,7 @@ if test "x$target_cpu" = xi386; then
   fi
 fi
 
-if test "x$target_cpu" = xi386 || test "x$target_cpu" = xx86_64; then
+if ( test "x$target_cpu" = xi386 || test "x$target_cpu" = xx86_64 ) && test "x$platform" != xemu; then
   # Some toolchains enable these features by default, but they need
   # registers that aren't set up properly in GRUB.
   TARGET_CFLAGS="$TARGET_CFLAGS -mno-mmx -mno-sse -mno-sse2 -mno-3dnow"
-- 
1.8.2.1

