From f333a71c6c02232151d9af98a8beae3b5e23e102 Mon Sep 17 00:00:00 2001
From: Colin Watson <cjwatson@ubuntu.com>
Date: Wed, 5 Sep 2012 13:55:54 +0100
Subject: [PATCH 031/364] * NEWS: Fix typo.

---
 ChangeLog | 4 ++++
 NEWS      | 2 +-
 2 files changed, 5 insertions(+), 1 deletion(-)

diff --git a/ChangeLog b/ChangeLog
index 9124825..9323887 100644
--- a/ChangeLog
+++ b/ChangeLog
@@ -1,5 +1,9 @@
 2012-09-05  Colin Watson  <cjwatson@ubuntu.com>
 
+	* NEWS: Fix typo.
+
+2012-09-05  Colin Watson  <cjwatson@ubuntu.com>
+
 	* util/import_gcry.py: Sort cipher_files, to make build system
 	generation more deterministic.
 
diff --git a/NEWS b/NEWS
index f9b06ab..4bb5f98 100644
--- a/NEWS
+++ b/NEWS
@@ -1,6 +1,6 @@
 New in 2.00:
 
-* Appearence:
+* Appearance:
   * Official theme for gfxmenu (starfield)
   * Menu is organised with submenus.
   * Better default video mode selection using EDID.
-- 
1.8.1.4

