From 4ad3b808b24202670c65a20e1f0a532499772312 Mon Sep 17 00:00:00 2001
From: Vladimir 'phcoder' Serbinenko <phcoder@gmail.com>
Date: Fri, 12 Apr 2013 01:44:14 +0200
Subject: [PATCH 283/364] 	* tests/partmap_test.in: Fix missing qemudisk
 setting.

---
 ChangeLog             | 4 ++++
 tests/partmap_test.in | 1 +
 2 files changed, 5 insertions(+)

diff --git a/ChangeLog b/ChangeLog
index c137b5f..0668a0f 100644
--- a/ChangeLog
+++ b/ChangeLog
@@ -1,3 +1,7 @@
+2013-04-12  Vladimir Serbinenko  <phcoder@gmail.com>
+
+	* tests/partmap_test.in: Fix missing qemudisk setting.
+
 2013-04-11  Vladimir Serbinenko  <phcoder@gmail.com>
 
 	Support i386-ieee1275 grub-mkrescue and make check on it.
diff --git a/tests/partmap_test.in b/tests/partmap_test.in
index a0beb2a..8d68a28 100644
--- a/tests/partmap_test.in
+++ b/tests/partmap_test.in
@@ -72,6 +72,7 @@ case "${grub_modinfo_target_cpu}-${grub_modinfo_platform}" in
 	;;
     *)
 	disk=hd0
+	qemudisk=hda
 	;;
 esac
 imgfile="`mktemp "${TMPDIR:-/tmp}/tmp.XXXXXXXXXX"`" || exit 1
-- 
1.8.1.4

