From 33f6759ef6669f4229e21296922bfad0bf9238c5 Mon Sep 17 00:00:00 2001
From: Vladimir 'phcoder' Serbinenko <phcoder@gmail.com>
Date: Sun, 14 Apr 2013 17:13:58 +0200
Subject: [PATCH 304/364] 	Support grub-shell on sparc64.

---
 ChangeLog                |  4 ++++
 tests/util/grub-shell.in | 10 ++++++++++
 2 files changed, 14 insertions(+)

diff --git a/ChangeLog b/ChangeLog
index 4c1e28d..867cc5d 100644
--- a/ChangeLog
+++ b/ChangeLog
@@ -1,5 +1,9 @@
 2013-04-14  Vladimir Serbinenko  <phcoder@gmail.com>
 
+	Support grub-shell on sparc64.
+
+2013-04-14  Vladimir Serbinenko  <phcoder@gmail.com>
+
 	Support mkrescue on sparc64.
 
 2013-04-14  Vladimir Serbinenko  <phcoder@gmail.com>
diff --git a/tests/util/grub-shell.in b/tests/util/grub-shell.in
index e467b4a..739c300 100644
--- a/tests/util/grub-shell.in
+++ b/tests/util/grub-shell.in
@@ -69,6 +69,16 @@ case "${grub_modinfo_target_cpu}-${grub_modinfo_platform}" in
 	serial_null="-serial null"
 	;;
 
+    sparc64-ieee1275)
+	boot=cd
+	qemu=qemu-system-sparc64
+	console=
+	serial_port=ieee1275/ttya
+	trim=1
+	qemuopts="$qemuopts -no-reboot"
+	halt_cmd=reboot
+	;;
+
     mips-qemu_mips)
 	boot=mips_qemu
 	qemu=qemu-system-mips
-- 
1.8.1.4

