From 3a36b768d2472c81b0f2b34f6e7fb41230b8a00a Mon Sep 17 00:00:00 2001
From: Vladimir 'phcoder' Serbinenko <phcoder@gmail.com>
Date: Sun, 14 Apr 2013 17:16:32 +0200
Subject: [PATCH 305/364] 	* tests/partmap_test.in: Skip on sparc64.

---
 ChangeLog             | 4 ++++
 tests/partmap_test.in | 6 ++++++
 2 files changed, 10 insertions(+)

diff --git a/ChangeLog b/ChangeLog
index 867cc5d..402c79f 100644
--- a/ChangeLog
+++ b/ChangeLog
@@ -1,5 +1,9 @@
 2013-04-14  Vladimir Serbinenko  <phcoder@gmail.com>
 
+	* tests/partmap_test.in: Skip on sparc64.
+
+2013-04-14  Vladimir Serbinenko  <phcoder@gmail.com>
+
 	Support grub-shell on sparc64.
 
 2013-04-14  Vladimir Serbinenko  <phcoder@gmail.com>
diff --git a/tests/partmap_test.in b/tests/partmap_test.in
index f667f86..bc503f5 100644
--- a/tests/partmap_test.in
+++ b/tests/partmap_test.in
@@ -66,6 +66,12 @@ case "${grub_modinfo_target_cpu}-${grub_modinfo_platform}" in
 	# QEMU firmware has bugs which prevent it from accessing hard disk.
 	exit 0
 	;;
+    sparc64-ieee1275)
+	disk=ieee1275//pci@1fe\,0/pci-ata@5/ide0@500/disk@0
+	qemudisk=hda
+	# QEMU firmware has bugs which prevent it from accessing hard disk.
+	exit 0
+	;;
     i386-ieee1275)
 	disk=ieee1275/d
 	qemudisk=hdb
-- 
1.8.1.4

