diff -ur b/preload.init.in a/preload.init.in
--- b/preload.init.in	2008-10-23 04:08:20.000000000 +0300
+++ a/preload.init.in	2013-04-22 03:45:35.967908041 +0300
@@ -5,6 +5,7 @@
 # Starts the @PACKAGE@ daemon
 #
 # chkconfig: 2345 05 95
+# chkconfig: 5 05 95
 # description: @PACKAGE_SUMMARY@
 # processname: @PACKAGE@
 #
@@ -12,7 +13,8 @@
 # Provides:          preload.sourceforge.net
 # Required-Start:    $local_fs $remote_fs $time
 # Required-Stop:     $local_fs $remote_fs $time
-# Default-Start:     2 3 4 5
+# Should-Start:      dm
+# Default-Start:     5
 # Default-Stop:      0 1 6
 # Short-Description: @PACKAGE_SUMMARY@
 # Description:       Analyzes what applications users run and tries to predict
@@ -24,6 +26,8 @@
 # Source function library.
 . /etc/rc.d/init.d/functions
 
+[ -f /dev/.readahead_collect ] && exit 0
+
 [ -x @sbindir@/@PACKAGE@ ] || exit 0
 
 if [ -f @sysconfigdir@/@PACKAGE@ ]; then
