--- Makefile.orig	2008-08-03 11:13:21.000000000 +0200
+++ Makefile	2008-08-03 11:18:22.000000000 +0200
@@ -14,9 +14,12 @@
 all:	$(LIBNAME) vstream-client
 
 install:	$(LIBNAME) vstream-client
-	$(INSTALL) -m 755 vstream-client $(BINDIR)/vstream-client
-	$(INSTALL) -m 644 $(LIBNAME) $(LIBDIR)/$(LIBNAME)
-	$(INSTALL) -m 644 vstream-client.h $(INCDIR)/vstream-client.h
+	$(INSTALL) -d $(D_BINDIR)
+	$(INSTALL) -m 755 vstream-client $(D_BINDIR)/vstream-client
+	$(INSTALL) -d $(D_LIBDIR)
+	$(INSTALL) -m 644 $(LIBNAME) $(D_LIBDIR)/$(LIBNAME)
+	$(INSTALL) -d $(D_INCDIR)
+	$(INSTALL) -m 644 vstream-client.h $(D_INCDIR)/vstream-client.h
 
 .c.o:
 	$(CC) -c $(CFLAGS) -o $@ $<
