--- /data/lightdm.conf	2012-10-01 04:57:50.000000000 +0300
+++ /data/lightdm.conf	2013-02-03 17:19:20.843238683 +0200
@@ -26,7 +26,7 @@
 #log-directory=/var/log/lightdm
 #run-directory=/var/run/lightdm
 #cache-directory=/var/cache/lightdm
-#xsessions-directory=/usr/share/xsessions
+xsessions-directory=/etc/X11/dm/Sessions
 #remote-sessions-directory=/usr/share/lightdm/remote-sessions
 #xgreeters-directory=/usr/share/xgreeters
 
@@ -77,7 +77,7 @@
 #user-session=default
 #allow-guest=true
 #guest-session=UNIMPLEMENTED
-#session-wrapper=lightdm-session
+session-wrapper=/etc/X11/Xsession
 #display-setup-script=
 #greeter-setup-script=
 #session-setup-script=