--- chrome/chrome_exe.gypi.ldflags	2012-06-28 09:02:24.000000000 -0400
+++ chrome/chrome_exe.gypi	2012-07-06 15:46:13.008003437 -0400
@@ -76,6 +76,14 @@
         'app/main_dll_loader_win.cc',
         'app/main_dll_loader_win.h',
       ],
+      # Hey, I know what would be an awesome idea.
+      # Let us statically compile EVERYTHING into one giant
+      # freaking executable. Too bad this means we can't build
+      # natively on an IA32 machine. If only someone would
+      # invent shared library objects. Idiots.
+      'ldflags': [
+        '-Wl,--no-keep-memory'
+      ],
       'mac_bundle_resources': [
         'app/app-Info.plist',
       ],
