Index: chromium-46.0.2490.71/chrome/test/data/webui_test_resources.grd
===================================================================
--- chromium-46.0.2490.71.orig/chrome/test/data/webui_test_resources.grd	2015-10-07 04:42:10.000000000 +0200
+++ chromium-46.0.2490.71/chrome/test/data/webui_test_resources.grd	2015-10-15 10:18:05.063915885 +0200
@@ -8,7 +8,6 @@
   </outputs>
   <release seq="1">
     <includes>
-      <include name="IDR_WEBUI_TEST_I18N_PROCESS_CSS_TEST" file="webui/i18n_process_css_test.html" flattenhtml="true" allowexternalscript="true" type="BINDATA" />
     </includes>
   </release>
 </grit>
