Index: chromium/src/third_party/skia/src/ports/SkFontHost_fontconfig.cpp
===================================================================
--- third_party/skia/src/ports/SkFontHost_fontconfig.cpp
+++ third_party/skia/src/ports/SkFontHost_fontconfig.cpp
@@ -13,6 +13,7 @@
 #include "SkTypeface.h"
 #include "SkTypefaceCache.h"
 #include "SkResourceCache.h"
+#include <unistd.h>
 
 ///////////////////////////////////////////////////////////////////////////////
 ///////////////////////////////////////////////////////////////////////////////

