--- coreutils-8.15/src/dircolors.hin.colors_mdkconf~	2012-01-01 10:04:06.000000000 +0100
+++ coreutils-8.15/src/dircolors.hin	2012-01-10 08:36:22.599206624 +0100
@@ -8,6 +8,10 @@
 # The keywords COLOR, OPTIONS, and EIGHTBIT (honored by the
 # slackware version of dircolors) are recognized but ignored.
 
+# This file goes in the /etc directory, and must be world readable.
+# You can copy this file to .dir_colors in your $HOME directory to override
+# the system defaults.
+
 # Below, there should be one TERM entry for each termtype that is colorizable
 TERM Eterm
 TERM ansi
@@ -81,7 +85,8 @@
 DOOR 01;35	# door
 BLK 40;33;01	# block device driver
 CHR 40;33;01	# character device driver
-ORPHAN 40;31;01 # symlink to nonexistent file, or non-stat'able file
+ORPHAN 01;05;37;41 # symlink to nonexistent file, or non-stat'able file
+MISSING 01;05;37;41 # ... and the files they point to
 SETUID 37;41	# file that is setuid (u+s)
 SETGID 30;43	# file that is setgid (g+s)
 CAPABILITY 30;41	# file with capability
@@ -97,15 +102,15 @@
 # (and any comments you want to add after a '#')
 
 # If you use DOS-style suffixes, you may want to uncomment the following:
-#.cmd 01;32 # executables (bright green)
-#.exe 01;32
-#.com 01;32
-#.btm 01;32
-#.bat 01;32
+.cmd 01;32 # executables (bright green)
+.exe 01;32
+.com 01;32
+.btm 01;32
+.bat 01;32
 # Or if you want to colorize scripts even if they do not have the
 # executable bit actually set.
-#.sh  01;32
-#.csh 01;32
+.sh  01;32
+.csh 01;32
 
  # archives or compressed (bright red)
 .tar 01;31
@@ -140,10 +145,13 @@
 .7z  01;31
 .rz  01;31
 .cab 01;31
+.iso 01;31
 
 # image formats
 .jpg 01;35
 .jpeg 01;35
+.JPG 01;35
+.JPEG 01;35
 .gif 01;35
 .bmp 01;35
 .pbm 01;35
@@ -155,6 +163,7 @@
 .tif 01;35
 .tiff 01;35
 .png 01;35
+.ico 01;35
 .svg 01;35
 .svgz 01;35
 .mng 01;35
@@ -166,6 +175,9 @@
 .mkv 01;35
 .webm 01;35
 .ogm 01;35
+.divx 01;35
+.xvid 01;35
+.3gp 01;35
 .mp4 01;35
 .m4v 01;35
 .mp4v 01;35
@@ -187,6 +199,7 @@
 .yuv 01;35
 .cgm 01;35
 .emf 01;35
+.gem 01;35
 
 # http://wiki.xiph.org/index.php/MIME_Types_and_File_Extensions
 .axv 01;35
@@ -212,3 +225,74 @@
 .oga 00;36
 .spx 00;36
 .xspf 00;36
+
+.mp2 00;36
+.mod 00;36
+.xm 00;36
+.s3m 00;36
+.it 00;36
+.wma 00;36
+
+# document and various text(ish) formats
+# .txt 01;37;40
+# .lyx 01;37;40
+# .doc 01;37;40
+# .wri 01;37;40
+# .odt 01;37;40
+# .wpd 01;37;40
+# .abw 01;37;40
+# .rtf 01;37;40
+# .sam 01;37;40
+# .pdb 01;37;40
+# .sxw 01;37;40
+# .kwd 01;37;40
+# .sgml 01;37;40
+# .html 01;37;40
+# .htm 01;37;40
+# .wml 01;37;40
+# .xml 01;37;40
+# .pdf 01;37;40
+# .ps 01;37;40
+# .man 01;37;40
+# .info 01;37;40
+# .tex 01;37;40
+# .conf 01;37;40
+# .cnf 01;37;40
+# .cfg 01;37;40
+# .config 01;37;40
+# .ini 01;37;40
+# .spec 01;37;40
+# .log 01;37;40
+# .c 01;37;40
+# .cpp 01;37;40
+# .cxx 01;37;40
+# .h 01;37;40
+# .hpp 01;37;40
+# .java 01;37;40
+# .pl 01;37;40
+# .py 01;37;40
+# .patch 01;37;40
+# *rc 01;37;40
+# *README 01;37;40
+# *COPYING 01;37;40
+# *ChangeLog 01;37;40
+# *NEWS 01;37;40
+# *AUTHORS 01;37;40
+# *ABOUT-NLS 01;37;40
+# *THANKS 01;37;40
+# *TODO 01;37;40
+# *FAQ 01;37;40
+# *INSTALL 01;37;40
+# *CHANGES 01;37;40
+
+# backup formats
+*~ 47;30
+.bak 47;30
+.swp 47;30
+.bck 47;30
+.bk 47;30
+.old 47;30
+.tmp 47;30
+.save 47;30
+.rpmsave 47;30
+.rpmnew 47;30
