--- a/files/usr/share/cinnamon/applets/calendar@cinnamon.org/applet.js
+++ b/files/usr/share/cinnamon/applets/calendar@cinnamon.org/applet.js
@@ -90,7 +90,7 @@ MyApplet.prototype = {
 
             // https://bugzilla.gnome.org/show_bug.cgi?id=655129
             this._upClient = new UPowerGlib.Client();
-            this._upClient.connect('notify-resume', this._updateClockAndDate);
+            this._upClient.connect('notify::resume', this._updateClockAndDate);
 
             // Start the clock
             this.on_settings_changed();