--- gcc-4.7.1/gcc/ada/gcc-interface/Makefile.in.orig	2012-07-10 17:33:42.146462985 +0200
+++ gcc-4.7.1/gcc/ada/gcc-interface/Makefile.in	2012-07-12 18:27:21.682409479 +0200
@@ -66,7 +66,7 @@
 # CFLAGS is for the user to override to, e.g., do a bootstrap with -O2.
 # BOOT_CFLAGS is the value of CFLAGS to pass
 # to the stage2 and stage3 compilations
-CFLAGS = -g
+CFLAGS =
 BOOT_CFLAGS = -O $(CFLAGS)
 # These exists to be overridden by the t-* files, respectively.
 T_CFLAGS =
@@ -105,12 +105,12 @@
 GNATBIND_FLAGS = -static -x
 ADA_CFLAGS =
 ADAFLAGS = -W -Wall -gnatpg -gnata
-FORCE_DEBUG_ADAFLAGS = -g
+FORCE_DEBUG_ADAFLAGS =
 NO_INLINE_ADAFLAGS = -fno-inline
 NO_SIBLING_ADAFLAGS = -fno-optimize-sibling-calls
 NO_REORDER_ADAFLAGS = -fno-toplevel-reorder
 GNATLIBFLAGS = -W -Wall -gnatpg -nostdinc
-GNATLIBCFLAGS = -g -O2
+GNATLIBCFLAGS = -O2
 PICFLAG_FOR_TARGET = @PICFLAG_FOR_TARGET@
 
 # Pretend that _Unwind_GetIPInfo is available for the target by default.  This
--- gcc-4.7.1/libada/Makefile.in.orig	2012-06-12 17:12:37.000000000 +0200
+++ gcc-4.7.1/libada/Makefile.in	2012-07-12 18:27:53.482408144 +0200
@@ -59,7 +59,7 @@ LDFLAGS=
 CFLAGS=-g
 PICFLAG = @PICFLAG@
 GNATLIBFLAGS= -W -Wall -gnatpg -nostdinc
-GNATLIBCFLAGS= -g -O2
+GNATLIBCFLAGS= -O2
 GNATLIBCFLAGS_FOR_C = -W -Wall $(GNATLIBCFLAGS) $(CFLAGS_FOR_TARGET) \
 	-fexceptions -DIN_RTS @have_getipinfo@
 
--- gcc-4.7.1/libffi/Makefile.am.orig	2012-02-27 11:00:12.000000000 +0100
+++ gcc-4.7.1/libffi/Makefile.am	2012-07-12 18:28:20.115740360 +0200
@@ -168,7 +168,7 @@
 
 LTLDFLAGS = $(shell $(SHELL) $(top_srcdir)/../libtool-ldflags $(LDFLAGS))
 
-AM_CFLAGS = -Wall -g -fexceptions
+AM_CFLAGS = -Wall -fexceptions
 if FFI_DEBUG
 # Build debug. Define FFI_DEBUG on the commandline so that, when building with
 # MSVC, it can link against the debug CRT.
--- gcc-4.7.1/libffi/Makefile.in.orig	2012-06-14 10:48:08.000000000 +0200
+++ gcc-4.7.1/libffi/Makefile.in	2012-07-12 18:28:35.029073067 +0200
@@ -475,7 +475,7 @@
 libffi_convenience_la_DEPENDENCIES = $(libffi_la_DEPENDENCIES)
 nodist_libffi_convenience_la_SOURCES = $(nodist_libffi_la_SOURCES)
 LTLDFLAGS = $(shell $(SHELL) $(top_srcdir)/../libtool-ldflags $(LDFLAGS))
-AM_CFLAGS = -Wall -g -fexceptions $(am__append_2)
+AM_CFLAGS = -Wall -fexceptions $(am__append_2)
 libffi_la_LDFLAGS = -no-undefined -version-info `grep -v '^\#' $(srcdir)/libtool-version` $(LTLDFLAGS) $(AM_LTLDFLAGS)
 AM_CPPFLAGS = -I. -I$(top_srcdir)/include -Iinclude -I$(top_srcdir)/src
 AM_CCASFLAGS = $(AM_CPPFLAGS)
--- gcc-4.7.1/libgcc/Makefile.in.orig	2012-05-24 16:59:38.000000000 +0200
+++ gcc-4.7.1/libgcc/Makefile.in	2012-07-12 18:28:59.042405393 +0200
@@ -221,7 +221,7 @@
 
 # Options to use when compiling libgcc2.a.
 #
-LIBGCC2_DEBUG_CFLAGS = -g
+LIBGCC2_DEBUG_CFLAGS =
 LIBGCC2_CFLAGS = -O2 $(LIBGCC2_INCLUDES) $(GCC_CFLAGS) $(HOST_LIBGCC2_CFLAGS) \
 		 $(LIBGCC2_DEBUG_CFLAGS) -DIN_LIBGCC2 \
 		 -fbuilding-libgcc -fno-stack-protector \
--- gcc-4.7.1/libjava/classpath/tools/Makefile.am.orig	2011-09-29 13:37:47.000000000 +0200
+++ gcc-4.7.1/libjava/classpath/tools/Makefile.am	2012-07-12 18:29:33.452403949 +0200
@@ -355,8 +355,8 @@
 if JAVA_MAINTAINER_MODE
 ## Compile ASM separately as it is latin-1 encoded.
 	AC=`echo $(JCOMPILER) | sed -e 's/UTF-8/ISO-8859-1/g'`; \
-	  $$AC -g -w -d $(srcdir)/asm @asm.lst
-	$(JCOMPILER) $(USE_JAVAC_FLAGS) -g -d $(srcdir)/classes @classes.lst @vm-tools.lst
+	  $$AC -w -d $(srcdir)/asm @asm.lst
+	$(JCOMPILER) $(USE_JAVAC_FLAGS) -d $(srcdir)/classes @classes.lst @vm-tools.lst
 endif
 ## END GCJ LOCAL
 ## Copy over tools resource files.
--- gcc-4.7.1/libjava/classpath/tools/Makefile.in.orig	2011-09-29 13:37:47.000000000 +0200
+++ gcc-4.7.1/libjava/classpath/tools/Makefile.in	2012-07-12 18:29:52.035736502 +0200
@@ -1413,8 +1413,8 @@
 	fi
 	cat classes.lst asm.lst vm-tools.lst > all-classes.lst
 @JAVA_MAINTAINER_MODE_TRUE@	AC=`echo $(JCOMPILER) | sed -e 's/UTF-8/ISO-8859-1/g'`; \
-@JAVA_MAINTAINER_MODE_TRUE@	  $$AC -g -w -d $(srcdir)/asm @asm.lst
-@JAVA_MAINTAINER_MODE_TRUE@	$(JCOMPILER) $(USE_JAVAC_FLAGS) -g -d $(srcdir)/classes @classes.lst @vm-tools.lst
+@JAVA_MAINTAINER_MODE_TRUE@	  $$AC -w -d $(srcdir)/asm @asm.lst
+@JAVA_MAINTAINER_MODE_TRUE@	$(JCOMPILER) $(USE_JAVAC_FLAGS) -d $(srcdir)/classes @classes.lst @vm-tools.lst
 	@list=`cd $(srcdir)/resource && find gnu/classpath/tools com/sun/tools/javac \
 	       sun/rmi/rmic $(GJDOC_EX) -name \*.properties -print -o -name \*.jav -print`; \
 	for p in $$list; do \
--- gcc-4.7.1/libjava/Makefile.am.orig	2011-08-05 16:37:48.000000000 +0200
+++ gcc-4.7.1/libjava/Makefile.am	2012-07-12 18:30:07.649069180 +0200
@@ -399,7 +399,7 @@
 ## Extra CFLAGS used for JNI C sources shared with GNU Classpath.
 PEDANTIC_CFLAGS = -ansi -pedantic -Wall -Wno-long-long
 
-JCFLAGS = -g
+JCFLAGS =
 JC1FLAGS = @LIBGCJ_JAVAFLAGS@ $(GCJFLAGS)
 
 LIBFFIINCS = @LIBFFIINCS@
--- gcc-4.7.1/libjava/Makefile.in.orig	2012-06-14 10:48:08.000000000 +0200
+++ gcc-4.7.1/libjava/Makefile.in	2012-07-12 18:30:20.642401968 +0200
@@ -1032,7 +1032,7 @@
 @BUILD_LIBGCJ_REDUCED_REFLECTION_FALSE@LIBGCJ_REDUCED_REFLECTION_FLAGS = 
 @BUILD_LIBGCJ_REDUCED_REFLECTION_TRUE@LIBGCJ_REDUCED_REFLECTION_FLAGS = -freduced-reflection
 PEDANTIC_CFLAGS = -ansi -pedantic -Wall -Wno-long-long
-JCFLAGS = -g
+JCFLAGS =
 JC1FLAGS = @LIBGCJ_JAVAFLAGS@ $(GCJFLAGS)
 AM_CPPFLAGS = -I$(top_srcdir) \
 	-Iinclude -I$(top_srcdir)/include \
--- gcc-4.7.1/libjava/testsuite/Makefile.am.orig	2010-05-04 00:37:50.000000000 +0200
+++ gcc-4.7.1/libjava/testsuite/Makefile.am	2012-07-12 18:30:54.965733863 +0200
@@ -83,11 +83,11 @@
 	  testtmpdir=`echo $$test | sed -e 's,/,_,g'`T; \
 	  rm -rf $$testtmpdir; $(mkdir_p) $$testtmpdir; \
 	  if test -n "$$testdep"; then \
-	    $(GCJ) -C -g -w --encoding=UTF-8 -bootclasspath \
+	    $(GCJ) -C -w --encoding=UTF-8 -bootclasspath \
 	      $(top_builddir)/libgcj-$(gcc_version).jar::$$testtmpdir \
 	      -d $$testtmpdir $(srcdir)/$$testdep || exit; \
 	  fi; \
-	  $(GCJ) -C -g -w --encoding=UTF-8 -bootclasspath \
+	  $(GCJ) -C -w --encoding=UTF-8 -bootclasspath \
 	    $(top_builddir)/libgcj-$(gcc_version).jar:$$testtmpdir \
 	    -d $$testtmpdir $(srcdir)/$$test || exit; \
 	  case "$$test" in \
--- gcc-4.7.1/libjava/testsuite/Makefile.in.orig	2012-05-21 20:14:01.000000000 +0200
+++ gcc-4.7.1/libjava/testsuite/Makefile.in	2012-07-12 18:31:09.359066590 +0200
@@ -569,11 +569,11 @@
 @JAVA_MAINTAINER_MODE_TRUE@	  testtmpdir=`echo $$test | sed -e 's,/,_,g'`T; \
 @JAVA_MAINTAINER_MODE_TRUE@	  rm -rf $$testtmpdir; $(mkdir_p) $$testtmpdir; \
 @JAVA_MAINTAINER_MODE_TRUE@	  if test -n "$$testdep"; then \
-@JAVA_MAINTAINER_MODE_TRUE@	    $(GCJ) -C -g -w --encoding=UTF-8 -bootclasspath \
+@JAVA_MAINTAINER_MODE_TRUE@	    $(GCJ) -C -w --encoding=UTF-8 -bootclasspath \
 @JAVA_MAINTAINER_MODE_TRUE@	      $(top_builddir)/libgcj-$(gcc_version).jar::$$testtmpdir \
 @JAVA_MAINTAINER_MODE_TRUE@	      -d $$testtmpdir $(srcdir)/$$testdep || exit; \
 @JAVA_MAINTAINER_MODE_TRUE@	  fi; \
-@JAVA_MAINTAINER_MODE_TRUE@	  $(GCJ) -C -g -w --encoding=UTF-8 -bootclasspath \
+@JAVA_MAINTAINER_MODE_TRUE@	  $(GCJ) -C -w --encoding=UTF-8 -bootclasspath \
 @JAVA_MAINTAINER_MODE_TRUE@	    $(top_builddir)/libgcj-$(gcc_version).jar:$$testtmpdir \
 @JAVA_MAINTAINER_MODE_TRUE@	    -d $$testtmpdir $(srcdir)/$$test || exit; \
 @JAVA_MAINTAINER_MODE_TRUE@	  case "$$test" in \
--- gcc-4.7.1/libstdc++-v3/include/Makefile.am.orig	2012-03-23 12:00:54.000000000 +0100
+++ gcc-4.7.1/libstdc++-v3/include/Makefile.am	2012-07-12 18:31:29.102399096 +0200
@@ -1154,22 +1154,22 @@
 # Build two precompiled C++ includes, stdc++.h.gch/*.gch
 ${pch1a_output}: ${allstamped} ${host_builddir}/c++config.h ${pch1_source}
 	-mkdir -p ${pch1_output_builddir}
-	$(CXX) $(PCHFLAGS) $(AM_CPPFLAGS) -O2 -g -std=gnu++0x ${pch1_source} \
+	$(CXX) $(PCHFLAGS) $(AM_CPPFLAGS) -O2 -std=gnu++0x ${pch1_source} \
 	-o $@
 
 ${pch1b_output}: ${allstamped} ${host_builddir}/c++config.h ${pch1_source}
 	-mkdir -p ${pch1_output_builddir}
-	$(CXX) $(PCHFLAGS) $(AM_CPPFLAGS) -O2 -g ${pch1_source} -o $@
+	$(CXX) $(PCHFLAGS) $(AM_CPPFLAGS) -O2 ${pch1_source} -o $@
 
 # Build a precompiled TR1 include, stdtr1c++.h.gch/O2.gch
 ${pch2_output}: ${pch2_source} ${pch1_output}
 	-mkdir -p ${pch2_output_builddir}
-	$(CXX) $(PCHFLAGS) $(AM_CPPFLAGS) -O2 -g ${pch2_source} -o $@
+	$(CXX) $(PCHFLAGS) $(AM_CPPFLAGS) -O2 ${pch2_source} -o $@
 
 # Build a precompiled extension include, extc++.h.gch/O2.gch
 ${pch3_output}: ${pch3_source} ${pch2_output}
 	-mkdir -p ${pch3_output_builddir}
-	$(CXX) $(PCHFLAGS) $(AM_CPPFLAGS) -O2 -g ${pch3_source} -o $@
+	$(CXX) $(PCHFLAGS) $(AM_CPPFLAGS) -O2 ${pch3_source} -o $@
 
 # For robustness sake (in light of junk files or in-source
 # configuration), copy from the build or source tree to the install
--- gcc-4.7.1/libstdc++-v3/include/Makefile.in.orig	2012-03-23 12:00:54.000000000 +0100
+++ gcc-4.7.1/libstdc++-v3/include/Makefile.in	2012-07-12 18:31:43.922398474 +0200
@@ -1543,22 +1543,22 @@
 # Build two precompiled C++ includes, stdc++.h.gch/*.gch
 ${pch1a_output}: ${allstamped} ${host_builddir}/c++config.h ${pch1_source}
 	-mkdir -p ${pch1_output_builddir}
-	$(CXX) $(PCHFLAGS) $(AM_CPPFLAGS) -O2 -g -std=gnu++0x ${pch1_source} \
+	$(CXX) $(PCHFLAGS) $(AM_CPPFLAGS) -O2 -std=gnu++0x ${pch1_source} \
 	-o $@
 
 ${pch1b_output}: ${allstamped} ${host_builddir}/c++config.h ${pch1_source}
 	-mkdir -p ${pch1_output_builddir}
-	$(CXX) $(PCHFLAGS) $(AM_CPPFLAGS) -O2 -g ${pch1_source} -o $@
+	$(CXX) $(PCHFLAGS) $(AM_CPPFLAGS) -O2 ${pch1_source} -o $@
 
 # Build a precompiled TR1 include, stdtr1c++.h.gch/O2.gch
 ${pch2_output}: ${pch2_source} ${pch1_output}
 	-mkdir -p ${pch2_output_builddir}
-	$(CXX) $(PCHFLAGS) $(AM_CPPFLAGS) -O2 -g ${pch2_source} -o $@
+	$(CXX) $(PCHFLAGS) $(AM_CPPFLAGS) -O2 ${pch2_source} -o $@
 
 # Build a precompiled extension include, extc++.h.gch/O2.gch
 ${pch3_output}: ${pch3_source} ${pch2_output}
 	-mkdir -p ${pch3_output_builddir}
-	$(CXX) $(PCHFLAGS) $(AM_CPPFLAGS) -O2 -g ${pch3_source} -o $@
+	$(CXX) $(PCHFLAGS) $(AM_CPPFLAGS) -O2 ${pch3_source} -o $@
 
 # For robustness sake (in light of junk files or in-source
 # configuration), copy from the build or source tree to the install
--- gcc-4.7.1/Makefile.in.orig	2012-05-16 17:54:24.000000000 +0200
+++ gcc-4.7.1/Makefile.in	2012-07-12 18:32:07.075730835 +0200
@@ -364,7 +364,7 @@
 
 # Flags to pass to stage2 and later makes.  They are defined
 # here so that they can be overridden by Makefile fragments.
-BOOT_CFLAGS= -g -O2
+BOOT_CFLAGS= -O2
 BOOT_LDFLAGS=
 BOOT_ADAFLAGS=-gnatpg -gnata
 
@@ -574,7 +574,7 @@
 LIBCFLAGS_FOR_TARGET = $(CFLAGS_FOR_TARGET)
 LIBCXXFLAGS_FOR_TARGET = $(CXXFLAGS_FOR_TARGET) -fno-implicit-templates
 LDFLAGS_FOR_TARGET = @LDFLAGS_FOR_TARGET@
-GOCFLAGS_FOR_TARGET = -O2 -g
+GOCFLAGS_FOR_TARGET = -O2
 
 FLAGS_FOR_TARGET = @FLAGS_FOR_TARGET@
 SYSROOT_CFLAGS_FOR_TARGET = @SYSROOT_CFLAGS_FOR_TARGET@ 
