--- a/panels/power/cc-power-panel.c
+++ b/panels/power/cc-power-panel.c
@@ -978,14 +978,17 @@ set_ac_battery_ui_mode (CcPowerPanel *se
 {
   gboolean has_batteries = FALSE;
   gboolean has_lid = FALSE;
+#if ! UP_CHECK_VERSION(0,99,0)
   gboolean ret;
   GError *error = NULL;
+#endif
   GPtrArray *devices;
   guint i;
   UpDevice *device;
   UpDeviceKind kind;
   CcPowerPanelPrivate *priv = self->priv;
 
+#if ! UP_CHECK_VERSION(0,99,0)
   /* this is sync, but it's cached in the daemon and so quick */
   ret = up_client_enumerate_devices_sync (self->priv->up_client, NULL, &error);
   if (!ret)
@@ -994,6 +997,7 @@ set_ac_battery_ui_mode (CcPowerPanel *se
       g_error_free (error);
       goto out;
     }
+#endif
 
   devices = up_client_get_devices (self->priv->up_client);
   for (i=0; i<devices->len; i++)
@@ -1013,7 +1017,9 @@ set_ac_battery_ui_mode (CcPowerPanel *se
 
   has_lid = up_client_get_lid_is_present (self->priv->up_client);
 
+#if ! UP_CHECK_VERSION(0,99,0)
 out:
+#endif
   gtk_widget_set_visible (WID (priv->builder, "combobox_lid_ac"), has_lid);
   gtk_widget_set_visible (WID (priv->builder, "label_lid_action"), has_lid);
   gtk_widget_set_visible (WID (priv->builder, "combobox_lid_battery"), has_batteries && has_lid);