--- a/panels/common/cc-common-language.c
+++ b/panels/common/cc-common-language.c
@@ -576,14 +576,12 @@
 
         ht = g_hash_table_new_full (g_str_hash, g_str_equal, g_free, g_free);
 
-#if 0
         /* Add some common regions */
         g_hash_table_insert (ht, g_strdup ("en_US.utf8"), g_strdup (_("United States")));
         g_hash_table_insert (ht, g_strdup ("de_DE.utf8"), g_strdup (_("Germany")));
         g_hash_table_insert (ht, g_strdup ("fr_FR.utf8"), g_strdup (_("France")));
         g_hash_table_insert (ht, g_strdup ("es_ES.utf8"), g_strdup (_("Spain")));
         g_hash_table_insert (ht, g_strdup ("zh_CN.utf8"), g_strdup (_("China")));
-#endif
 
         gdm_parse_language_name (lang, &language, NULL, NULL, NULL);
         langs = gdm_get_all_language_names ();
--- a/panels/region/cinnamon-region-panel.ui
+++ b/panels/region/cinnamon-region-panel.ui
@@ -318,21 +318,6 @@
                                 <property name="homogeneous">True</property>
                               </packing>
                             </child>
-                            <child>
-                              <object class="GtkToolButton" id="region_remove">
-                                <property name="use_action_appearance">False</property>
-                                <property name="visible">True</property>
-                                <property name="label" translatable="yes">Remove Region</property>
-                                <property name="sensitive">False</property>
-                                <property name="can_focus">False</property>
-                                <property name="use_underline">True</property>
-                                <property name="icon_name">list-remove-symbolic</property>
-                              </object>
-                              <packing>
-                                <property name="expand">False</property>
-                                <property name="homogeneous">True</property>
-                              </packing>
-                            </child>
                           </object>
                           <packing>
                             <property name="expand">False</property>
@@ -806,25 +791,6 @@
                                       <object class="GtkBox" id="i_s_sp_box">
                                         <property name="visible">True</property>
                                         <child>
-                                           <object class="GtkButton" id="input_source_settings">
-                                             <property name="visible">True</property>
-                                             <child internal-child="accessible">
-                                               <object class="AtkObject" id="i_s_s_a11y">
-                                                 <property name="accessible-name" translatable="yes">Input Source Settings</property>
-                                               </object>
-                                             </child>
-                                             <child>
-                                                <object class="GtkImage" id="i_s_s_image">
-
-                                                  <property name="visible">True</property>
-                                                  <property name="icon_name">preferences-system-symbolic</property>
-                                                  <property name="icon_size">1</property>
-                                                  <property name="pixel_size">16</property>
-                                                </object>
-                                             </child>
-                                           </object>
-                                        </child>
-                                        <child>
                                            <object class="GtkButton" id="input_source_show">
                                              <property name="visible">True</property>
                                              <child internal-child="accessible">