--- openjdk/hotspot/src/os_cpu/linux_zero/vm/globals_linux_zero.hpp	2012-03-23 10:57:01.000000000 -0400
+++ openjdk/hotspot/src/os_cpu/linux_zero/vm/globals_linux_zero.hpp	2012-03-23 10:57:00.000000000 -0400
@@ -34,9 +34,9 @@
 define_pd_global(bool,  DontYieldALot,           false);
 define_pd_global(intx,  ThreadStackSize,         1664);
 #ifdef _LP64
-define_pd_global(intx,  VMThreadStackSize,       1024);
+define_pd_global(intx,  VMThreadStackSize,       1664);
 #else
-define_pd_global(intx,  VMThreadStackSize,       512);
+define_pd_global(intx,  VMThreadStackSize,       1152);
 #endif // _LP64
 define_pd_global(intx,  CompilerThreadStackSize, 0);
 define_pd_global(uintx, JVMInvokeMethodSlack,    8192);
