From d20dff1409fc00e783519cfc0065c6311bb914fb Mon Sep 17 00:00:00 2001
From: Matúš Kukan <matus.kukan@collabora.com>
Date: Tue, 03 Dec 2013 07:23:16 +0000
Subject: Package cppuhelper_unorc does not exist.

Change-Id: Ib80cdc1408c8784db066b35f27993f8521772a76
---
diff --git a/RepositoryExternal.mk b/RepositoryExternal.mk
index 3073c565..b44927e 100644
--- a/RepositoryExternal.mk
+++ b/RepositoryExternal.mk
@@ -3078,7 +3078,6 @@ endef
 define gb_Executable__register_climaker
 $(call gb_Executable_add_runtime_dependencies,climaker,\
 	$(call gb_Library_get_target,$(gb_CPPU_ENV)_uno) \
-	$(call gb_Package_get_target_for_build,cppuhelper_unorc) \
 	$(call gb_Rdb_get_target_for_build,ure/services) \
 	$(INSTROOT)/$(LIBO_URE_SHARE_FOLDER)/misc/services.rdb \
 	$(call gb_UnoApi_get_target,udkapi) \
@@ -3112,7 +3111,6 @@ $(call gb_Executable_add_runtime_dependencies,gengal,\
 	$(if $(filter-out MACOSX WNT,$(OS_FOR_BUILD)),$(if $(ENABLE_HEADLESS),, \
 		$(call gb_Library_get_target,vclplug_svp) \
 	)) \
-	$(call gb_Package_get_target_for_build,cppuhelper_unorc) \
 	$(call gb_Package_get_target_for_build,postprocess_registry) \
 	$(call gb_Rdb_get_target_for_build,ure/services) \
 	$(INSTROOT)/$(LIBO_URE_SHARE_FOLDER)/misc/services.rdb \
@@ -3152,7 +3150,6 @@ endef
 define gb_Executable__register_saxparser
 $(call gb_Executable_add_runtime_dependencies,saxparser,\
 	$(call gb_Library_get_target,$(gb_CPPU_ENV)_uno) \
-	$(call gb_Package_get_target_for_build,cppuhelper_unorc) \
 	$(call gb_Rdb_get_target_for_build,saxparser) \
 	$(call gb_Rdb_get_target_for_build,ure/services) \
 	$(INSTROOT)/$(LIBO_URE_SHARE_FOLDER)/misc/services.rdb \
@@ -3167,7 +3164,6 @@ endef
 define gb_Executable__register_uno
 $(call gb_Executable_add_runtime_dependencies,uno,\
 	$(call gb_Library_get_target,$(gb_CPPU_ENV)_uno) \
-	$(call gb_Package_get_target_for_build,cppuhelper_unorc) \
 	$(call gb_Rdb_get_target_for_build,ure/services) \
 	$(INSTROOT)/$(LIBO_URE_SHARE_FOLDER)/misc/services.rdb \
 	$(call gb_UnoApi_get_target,udkapi) \
diff --git a/solenv/gbuild/CppunitTest.mk b/solenv/gbuild/CppunitTest.mk
index 993757d..b3d6c96 100644
--- a/solenv/gbuild/CppunitTest.mk
+++ b/solenv/gbuild/CppunitTest.mk
@@ -156,7 +156,6 @@ $(call gb_CppunitTest_get_target,$(1)) : $(call gb_Library_get_target,$(gb_CPPU_
 $(call gb_CppunitTest_get_target,$(1)) : $(call gb_Library_get_target,affine_uno)
 $(call gb_CppunitTest_get_target,$(1)) : $(call gb_Library_get_target,unobootstrapprotector)
 $(call gb_CppunitTest_get_target,$(1)) : $(call gb_Library_get_target,unoexceptionprotector)
-$(call gb_CppunitTest_get_target,$(1)) : $(call gb_Package_get_target,cppuhelper_unorc)
 
 endef
 
--
cgit v0.9.0.2-2-gbebe
