--- gcstar/lib/gcstar/GCLang/TR/GCModels/GCbooks.pm	2013-09-06 10:12:48.116000045 +0300
+++ gcstar/lib/gcstar/GCLang/TR/GCModels/GCbooks.pm	2013-08-30 23:23:24.218295149 +0300
@@ -0,0 +1,136 @@
+{
+    package GCLang::TR::GCModels::GCbooks;
+
+    use utf8;
+###################################################
+#
+#  Copyright 2005-2010 Christian Jodar
+#
+#  This file is part of GCstar.
+#
+#  GCstar is free software; you can redistribute it and/or modify
+#  it under the terms of the GNU General Public License as published by
+#  the Free Software Foundation; either version 2 of the License, or
+#  (at your option) any later version.
+#
+#  GCstar is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#  GNU General Public License for more details.
+#
+#  You should have received a copy of the GNU General Public License
+#  along with GCstar; if not, write to the Free Software
+#  Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+#
+###################################################
+    
+    use strict;
+    use base 'Exporter';
+
+    our @EXPORT = qw(%lang);
+
+    our %lang = (
+    
+        CollectionDescription => 'Kitap Arşivleri',
+        Items => 'Kitaplar',
+        NewItem => 'Yeni Kitap',
+    
+        Isbn => 'ISBN',
+        Title => 'Başlık',
+        Cover => 'Kapak',
+        Authors => 'Yazar',
+        Publisher => 'Yayıncı',
+        Publication => 'Yayımlama tarihi',
+        Language => 'Dil',
+        Genre => 'Tür',
+        Serie => 'Arşiv',
+        Rank => 'Sıralama',
+        Bookdescription => 'Açıklama',
+        Pages => 'Sayfa',
+        Read => 'Okundu',
+        Acquisition => 'Edinme tarihi',
+        Edition => 'Basım',
+        Format => 'Biçim',
+        Comments => 'Yorumlar',
+        Url => 'Web sayfası',
+        Translator => 'Çevirmen',
+        Artist => 'Sanatçı',
+        DigitalFile => 'Sayısal sürüm',
+
+        General => 'Genel',
+        Details => 'Ayrıntılar',
+
+        ReadNo => 'Okunmadı',
+        ReadYes => 'Okundu',
+     );
+}
+
+1;
+{
+    package GCLang::TR::GCModels::GCbooks;
+
+    use utf8;
+###################################################
+#
+#  Copyright 2005-2010 Christian Jodar
+#
+#  This file is part of GCstar.
+#
+#  GCstar is free software; you can redistribute it and/or modify
+#  it under the terms of the GNU General Public License as published by
+#  the Free Software Foundation; either version 2 of the License, or
+#  (at your option) any later version.
+#
+#  GCstar is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#  GNU General Public License for more details.
+#
+#  You should have received a copy of the GNU General Public License
+#  along with GCstar; if not, write to the Free Software
+#  Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301, USA
+#
+###################################################
+    
+    use strict;
+    use base 'Exporter';
+
+    our @EXPORT = qw(%lang);
+
+    our %lang = (
+    
+        CollectionDescription => 'Kitap Arşivleri',
+        Items => 'Kitaplar',
+        NewItem => 'Yeni Kitap',
+    
+        Isbn => 'ISBN',
+        Title => 'Başlık',
+        Cover => 'Kapak',
+        Authors => 'Yazarlar',
+        Publisher => 'Yayıncı',
+        Publication => 'Yayın Tarihi',
+        Language => 'Dil',
+        Genre => 'Tür',
+        Serie => 'Arşiv',
+        Rank => 'Sıralama',
+        Bookdescription => 'Tanım',
+        Pages => 'Sayfa',
+        Read => 'Oku',
+        Acquisition => 'Edinme tarihi',
+        Edition => 'Basım',
+        Format => 'Biçim',
+        Comments => 'Yorumlar',
+        Url => 'Web sayfası',
+        Translator => 'Çevirmen',
+        Artist => 'Sanatçı',
+        DigitalFile => 'Sayısal sürüm',
+
+        General => 'Genel',
+        Details => 'Detaylar',
+
+        ReadNo => 'Okunmadı',
+        ReadYes => 'Okundu',
+     );
+}
+
+1;
