--- gcstar/lib/gcstar/GCLang/TR/GCModels/GCboardgames.pm	2013-09-06 10:12:48.116000045 +0300
+++ gcstar/lib/gcstar/GCLang/TR/GCModels/GCboardgames.pm	2013-09-06 13:33:47.338489699 +0300
@@ -0,0 +1,86 @@
+{
+    package GCLang::TR::GCModels::GCboardgames;
+
+    use utf8;
+###################################################
+#
+#  Copyright 2005-2010 Christian Jodar
+#
+#  This file is part of GCstar.
+#
+#  GCstar is free software; you can redistribute it and/or modify
+#  it under the terms of the GNU General Public License as published by
+#  the Free Software Foundation; either version 2 of the License, or
+#  (at your option) any later version.
+#
+#  GCstar is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#  GNU General Public License for more details.
+#
+#  You should have received a copy of the GNU General Public License
+#  along with GCstar; if not, write to the Free Software
+#  Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301, USA
+#
+###################################################
+    
+    use strict;
+    use base 'Exporter';
+
+    our @EXPORT = qw(%lang);
+
+    our %lang = (
+    
+        CollectionDescription => 'Tahta oyunları arşivi',
+        Items => {0 => 'Game',
+                  1 => 'Game',
+                  X => 'Games'},
+        NewItem => 'Yeni oyun',
+    
+        Id => 'Id',
+        Name => 'Adı',
+        Original => 'Özgün adı',
+        Box => 'Kapak resmi',
+        DesignedBy => 'Tasarlayan',
+        PublishedBy => 'Yayınlayan',
+        Players => 'Oyuncu sayısı',
+        PlayingTime => 'Oynama zamanı',
+        SuggestedAge => 'Önerilen yaş',
+        Released => 'Dağıtım',
+        Description => 'Tanım',
+        Category => 'Grup',
+        Mechanics => 'Mekanik',
+        ExpandedBy => 'Genişleten',
+        ExpansionFor => 'Genişleme',
+        GameFamily => 'Oyun ailesi',
+        IllustratedBy => 'Anlatan',
+        Url => 'Web sayfası',
+        TimesPlayed => 'Oynanma sayısı',
+        CompleteContents => 'İçeriği tamamla',
+        Copies => 'Kopya numarası',
+        Condition => 'Koşul',
+        Photos => 'Resimler',
+        Photo1 => 'İlk resim',
+        Photo2 => 'İkinci resim',
+        Photo3 => 'Üçüncü resim',
+        Photo4 => 'Dördüncü resim',
+        Comments => 'Yorumlar',
+
+        Perfect => 'Çok İyi',
+        Good => 'İyi',
+        Average => 'Orta',
+        Poor => 'Zayıf',
+
+        CompleteYes => 'İçeriği tamamla',
+        CompleteNo => 'Kayıp parçalar',
+
+        General => 'Genel',
+        Details => 'Detaylar',
+        Personal => 'Kişisel',
+        Information => 'Bilgi',
+
+        FilterRatingSelect => 'Panı En Az...',
+     );
+}
+
+1;
