--- gcstar/lib/gcstar/GCLang/TR/GCModels/GCcomics.pm	2013-09-08 21:14:54.256000028 +0300
+++ gcstar/lib/gcstar/GCLang/TR/GCModels/GCcomics.pm	2013-08-30 23:41:41.347320182 +0300
@@ -0,0 +1,73 @@
+{
+    package GCLang::TR::GCModels::GCcomics;
+
+    use utf8;
+###################################################
+#
+#  Copyright 2005-2010 Christian Jodar
+#
+#  This file is part of GCstar.
+#
+#  GCstar is free software; you can redistribute it and/or modify
+#  it under the terms of the GNU General Public License as published by
+#  the Free Software Foundation; either version 2 of the License, or
+#  (at your option) any later version.
+#
+#  GCstar is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#  GNU General Public License for more details.
+#
+#  You should have received a copy of the GNU General Public License
+#  along with GCstar; if not, write to the Free Software
+#  Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301, USA
+#
+###################################################
+    
+    use strict;
+    use base 'Exporter';
+
+    our @EXPORT = qw(%lang);
+
+    our %lang = (
+    
+        CollectionDescription => 'Karikatür arşivi',
+        Items => {0 => 'Comics',
+                  1 => 'Comic',
+                  X => 'Comics'},
+        NewItem => 'Yeni karikatür',
+    
+    
+        Id => 'Id',
+        Name => 'Adı',
+        Series => 'Seri',
+        Volume => 'Cilt',
+        Title => 'Başlık',
+        Writer => 'Yazar',
+        Illustrator => 'Çizen',
+        Colourist => 'Renklendiren',
+        Publisher => 'Yayıncı',
+        Synopsis => 'Özet',
+        Collection => 'Arşiv',
+        PublishDate => 'Yayın Tarihi',
+        PrintingDate => 'Baskı Tarihi',
+        ISBN => 'ISBN',
+        Type => 'Tür',
+		Category => 'Grup',
+        Format => 'Biçim',
+        NumberBoards => 'Sayfa sayısı',
+		Signing => 'İmza',
+        Cost => 'Ücret',
+        Rating => 'Puan',
+        Comment => 'Yorumlar',
+        Url => 'Web sayfası',
+
+        FilterRatingSelect => 'Puanı En Az...',
+
+        Main => 'Ana Ögeler',
+        General => 'Genel',
+        Details => 'Detaylar',
+     );
+}
+
+1;
