--- gcstar/lib/gcstar/GCLang/TR/GCModels/GCmusics.pm	2013-09-08 21:14:54.256000028 +0300
+++ gcstar/lib/gcstar/GCLang/TR/GCModels/GCmusics.pm	2013-08-30 23:49:42.330331157 +0300
@@ -0,0 +1,66 @@
+{
+    package GCLang::TR::GCModels::GCmusics;
+
+    use utf8;
+###################################################
+#
+#  Copyright 2005-2010 Christian Jodar
+#
+#  This file is part of GCstar.
+#
+#  GCstar is free software; you can redistribute it and/or modify
+#  it under the terms of the GNU General Public License as published by
+#  the Free Software Foundation; either version 2 of the License, or
+#  (at your option) any later version.
+#
+#  GCstar is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#  GNU General Public License for more details.
+#
+#  You should have received a copy of the GNU General Public License
+#  along with GCstar; if not, write to the Free Software
+#  Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301, USA
+#
+###################################################
+    
+    use strict;
+    use base 'Exporter';
+
+    our @EXPORT = qw(%lang);
+
+    our %lang = (
+    
+        CollectionDescription => 'Müzik Arşivi',
+        Items => 'Albümler',
+        NewItem => 'Yeni Albüm',
+    
+        Unique => 'ISRC/EAN',
+        Title => 'Başlık',
+        Cover => 'Kapak',
+        Artist => 'Sanatçı',
+        Format => 'Biçim',
+        Running => 'Çalma Süresi',
+        Release => 'Dağıtım tarihi',
+        Genre => 'Tür',
+        Origin => 'Kaynak',
+
+#For tracks list
+        Tracks => 'Parça Listesi',
+        Number => 'Nu.',
+        Track => 'Başlık',
+        Time => 'Zaman',
+
+        Composer => 'Besteci',
+        Producer => 'Yapımcı',
+        Playlist => 'Çalma Listesi',
+        Comments => 'Yorumlar',
+        Label => 'Etiket',
+        Url => 'Web sayfası',
+
+        General => 'Genel',
+        Details => 'Detaylar',
+     );
+}
+
+1;
