--- gcstar/lib/gcstar/GCLang/TR/GCModels/GCsoftware.pm	2013-09-08 21:14:54.256000028 +0300
+++ gcstar/lib/gcstar/GCLang/TR/GCModels/GCsoftware.pm	2013-09-06 13:36:26.048496143 +0300
@@ -0,0 +1,84 @@
+{
+    package GCLang::TR::GCModels::GCsoftware;
+
+    use utf8;
+###################################################
+#
+#  Copyright 2005-2009 Tian
+#
+#  This file is part of GCstar.
+#
+#  GCstar is free software; you can redistribute it and/or modify
+#  it under the terms of the GNU General Public License as published by
+#  the Free Software Foundation; either version 2 of the License, or
+#  (at your option) any later version.
+#
+#  GCstar is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#  GNU General Public License for more details.
+#
+#  You should have received a copy of the GNU General Public License
+#  along with GCstar; if not, write to the Free Software
+#  Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301, USA
+#
+###################################################
+    
+    use strict;
+    use base 'Exporter';
+
+    our @EXPORT = qw(%lang);
+
+    our %lang = (
+    
+        CollectionDescription => 'Bilgisayar yazılım arşivi',
+        Items => {0 => 'Application',
+                  1 => 'Application',
+                  X => 'Applications',
+                  lowercase1 => 'application',
+                  lowercaseX => 'applications'},
+        NewItem => 'Yeni Uygulama',
+    
+        Id => 'Id',
+        Ean => 'EAN',
+        Name => 'Adı',
+        Platform => 'Platform',
+        Released => 'Dağıtım Tarihi',
+        Homepage => 'Ana Sayfası',		
+        Editor => 'Düzenleyen',
+        Developer => 'Geliştirici',
+        Category => 'Grup',
+		NumberOfCopies => 'Kopyalar',
+		Price => 'Ücret',
+        Box => 'Kapak resmi',
+        Case => 'Durum',
+        Manual => 'El Klavuzu',
+        Executable => 'Yürütülebilir',
+        Description => 'Açıklama',
+        License => 'Lisans',
+        Commercial => 'Ticari',
+		FreewareNoncommercial => 'Özgür (ticari olmayan)',		
+		OtherOpenSource => 'Diğer Açık Kaynak',
+		PublicDomain => 'Kamu Malı',
+		OtherLicense => 'Diğer',
+		Registration => 'Kayıt',				
+		RegistrationInfo => 'Kayıt Bilgisi',		
+		RegInfo => 'Kayıt Bilgisi',
+		RegistrationName => 'Kullanıcı adı',
+		RegistrationNumber => 'Kayıt Numarası',		
+		PanelRegistration => 'Kayıt Bilgisi',	
+		RegistrationComments => 'Ek bilgi veya yorumlar',
+        Screenshots => 'Ekran Görüntüsü',
+        Screenshot1 => 'İlk Ekran Görüntüsü',
+        Screenshot2 => 'İkinci Ekran Görüntüsü',
+        Comments => 'Yorumlar',
+        Url => 'Web sayfası',
+        General => 'Genel',
+        Details => 'Detaylar',
+        Information => 'Bilgi',
+
+        FilterRatingSelect => 'En Az Puan...',
+     );
+}
+
+1;
