--- gcstar-orig/bin/gcstar	2009-11-19 21:31:24.000000000 +0100
+++ gcstar/bin/gcstar	2009-11-20 22:36:58.731092949 +0100
@@ -40,17 +40,12 @@
 use Getopt::Long;
 use File::Path;
 use File::Spec;
-use FindBin qw($RealBin);
 use POSIX qw(locale_h);
 
-$ENV{GCS_BIN_DIR} = $RealBin;
-($ENV{GCS_LIB_DIR} = $ENV{GCS_BIN_DIR}) =~ s/bin\/?$/lib\/gcstar/;
-($ENV{GCS_SHARE_DIR} = $ENV{GCS_BIN_DIR}) =~ s/bin\/?$/share\/gcstar/;
-use lib File::Spec->canonpath("$RealBin/../lib/gcstar");
-if (exists $ENV{PAR_TEMP})
-{
-    unshift @INC, "$RealBin/../lib/gcstar";
-}
+use lib qw(/usr/share/gcstar/lib);
+$ENV{GCS_BIN_DIR} = '/usr/bin';
+$ENV{GCS_LIB_DIR} = '/usr/share/gcstar/lib';
+$ENV{GCS_SHARE_DIR} = '/usr/share/gcstar';
 
 # For win32, redirect all the output to files
 if ($^O =~ /win32/i)
@@ -80,23 +75,24 @@
 $ENV{GCS_CONFIG_HOME} = $ENV{XDG_CONFIG_HOME} if ($^O =~ /win32/i);
 
 # Migrate settings from old windows data path if required
-if (($^O =~ /win32/i) && !(-d $ENV{XDG_DATA_HOME}))
-{
-    if (-d $RealBin.'/../config')
-    {	
-        print ("Need to migrate settings from old gcstar data store\n");
-		mkpath $ENV{XDG_CONFIG_HOME};
-        mkpath $ENV{XDG_DATA_HOME};
-		my $winRealBin = $RealBin;
-		$winRealBin =~ s/\//\\/g;
-		my $winXDGConfig = $ENV{XDG_CONFIG_HOME};
-		$winXDGConfig =~ s/\//\\/g;
-		my $winXDGData = $ENV{XDG_CONFIG_HOME};
-		$winXDGData =~ s/\//\\/g;
-		system "xcopy /Y \"".$winRealBin."\\..\\config\" \"".$winXDGConfig."\"";
-		system "xcopy /Y \"".$winRealBin."\\..\\data\" \"".$winXDGData."\"";
-    } 
-}
+# we don't need it, we are using linux (uzsolt)
+# if (($^O =~ /win32/i) && !(-d $ENV{XDG_DATA_HOME}))
+# {
+#     if (-d $RealBin.'/../config')
+#     {	
+#         print ("Need to migrate settings from old gcstar data store\n");
+# 		mkpath $ENV{XDG_CONFIG_HOME};
+#         mkpath $ENV{XDG_DATA_HOME};
+# 		my $winRealBin = $RealBin;
+# 		$winRealBin =~ s/\//\\/g;
+# 		my $winXDGConfig = $ENV{XDG_CONFIG_HOME};
+# 		$winXDGConfig =~ s/\//\\/g;
+# 		my $winXDGData = $ENV{XDG_CONFIG_HOME};
+# 		$winXDGData =~ s/\//\\/g;
+# 		system "xcopy /Y \"".$winRealBin."\\..\\config\" \"".$winXDGConfig."\"";
+# 		system "xcopy /Y \"".$winRealBin."\\..\\data\" \"".$winXDGData."\"";
+#     } 
+# }
 
 mkpath $ENV{XDG_CONFIG_HOME};
 mkpath $ENV{XDG_DATA_HOME};
--- gcstar-orig/install	2009-11-19 20:59:19.000000000 +0100
+++ gcstar/install	2009-11-20 22:24:55.315124623 +0100
@@ -256,7 +256,7 @@
 
     #Copying lib
     verbosePrint $lang{InstallCopyDirectory}, $baseDir.'/lib/gcstar';
-    recursiveCopy('lib/gcstar', $baseDir.'/lib/gcstar');
+    recursiveCopy('lib/gcstar', $baseDir.'/share/gcstar/lib');
 
     #Copying share
     verbosePrint $lang{InstallCopyDirectory}, $baseDir.'/share/gcstar'; 
