diff -up qtwebkit-opensource-src-5.0.2/Source/WebCore/Target.pri.system_angle qtwebkit-opensource-src-5.0.2/Source/WebCore/Target.pri
--- qtwebkit-opensource-src-5.0.2/Source/WebCore/Target.pri.system_angle	2013-04-08 19:11:51.000000000 -0500
+++ Source/WebCore/Target.pri	2013-05-25 07:24:57.722118644 -0500
@@ -3988,7 +3988,7 @@ use?(3D_GRAPHICS) {
         platform/graphics/texmap/TextureMapperGL.cpp \
         platform/graphics/texmap/TextureMapperShaderManager.cpp
 
-    INCLUDEPATH += $$PWD/platform/graphics/gpu
+    INCLUDEPATH += $$PWD/platform/graphics/gpu /usr/include/GLSLANG
 
     contains(QT_CONFIG, opengl) | contains(QT_CONFIG, opengles2) {
         !contains(QT_CONFIG, opengles2) {
diff -up qtwebkit-opensource-src-5.0.2/Source/WebCore/WebCore.pri.system_angle qtwebkit-opensource-src-5.0.2/Source/WebCore/WebCore.pri
--- qtwebkit-opensource-src-5.0.2/Source/WebCore/WebCore.pri.system_angle	2013-05-25 07:24:57.718118687 -0500
+++ Source/WebCore/WebCore.pri	2013-05-25 09:24:18.318809566 -0500
@@ -214,12 +214,13 @@ enable?(WEB_AUDIO) {
 
 use?(3D_GRAPHICS) {
     win32: {
-        win32-g++: {
+        win32-g++*: {
             # Make sure OpenGL libs are after the webcore lib so MinGW can resolve symbols
             contains(QT_CONFIG, opengles2) {
                 CONFIG(debug, debug|release):contains(QT_CONFIG, angle) {
                     LIBS += $$QMAKE_LIBS_OPENGL_ES2_DEBUG
                 } else {
+                    INCLUDEPATH += /usr/i686-w64-mingw32/sys-root/mingw/include/GLSLANG
                     LIBS += $$QMAKE_LIBS_OPENGL_ES2
                 }
             } else {
@@ -228,6 +229,8 @@ use?(3D_GRAPHICS) {
         }
     } else {
         contains(QT_CONFIG, opengles2): LIBS += -lEGL
+        INCLUDEPATH += /usr/include/GLSLANG 
+        LIBS += -ltranslator_glsl -ltranslator_common -lpreprocessor
     }
 }
 
diff -up qtwebkit-opensource-src-5.0.2/Source/WebKit2/Target.pri.system_angle qtwebkit-opensource-src-5.0.2/Source/WebKit2/Target.pri
--- qtwebkit-opensource-src-5.0.2/Source/WebKit2/Target.pri.system_angle	2013-04-08 19:11:42.000000000 -0500
+++ Source/WebKit2/Target.pri	2013-05-25 07:24:57.723118634 -0500
@@ -938,7 +938,10 @@ enable?(TOUCH_EVENTS) {
 
 enable?(GEOLOCATION): QT += location
 
-use?(3D_GRAPHICS): WEBKIT += angle
+use?(3D_GRAPHICS) {
+    WEBKIT += angle
+    INCLUDEPATH += /usr/include/GLSLANG
+}
 
 use?(PLUGIN_BACKEND_XLIB) {
     DEFINES += XP_UNIX
diff -up qtwebkit-opensource-src-5.0.2/Source/WebKit/WebKit1.pro.system_angle qtwebkit-opensource-src-5.0.2/Source/WebKit/WebKit1.pro
--- qtwebkit-opensource-src-5.0.2/Source/WebKit/WebKit1.pro.system_angle	2013-04-08 19:11:39.000000000 -0500
+++ Source/WebKit/WebKit1.pro	2013-05-25 07:24:57.722118644 -0500
@@ -92,7 +92,10 @@ HEADERS += \
 INCLUDEPATH += \
     $$PWD/qt/WebCoreSupport
 
-use?(3D_GRAPHICS): WEBKIT += angle
+use?(3D_GRAPHICS) {
+    WEBKIT += angle
+    INCLUDEPATH += /usr/include/GLSLANG
+}
 
 enable?(GEOLOCATION) {
      HEADERS += \
diff -up qtwebkit-opensource-src-5.0.2/Tools/qmake/mkspecs/modules/angle.prf.system_angle qtwebkit-opensource-src-5.0.2/Tools/qmake/mkspecs/modules/angle.prf
diff -up qtwebkit-opensource-src-5.0.2/WebKit.pro.system_angle qtwebkit-opensource-src-5.0.2/WebKit.pro
--- qtwebkit-opensource-src-5.0.2/WebKit.pro.system_angle	2013-04-08 19:11:32.000000000 -0500
+++ WebKit.pro	2013-05-25 07:24:57.720118666 -0500
@@ -18,11 +18,11 @@ JavaScriptCore.file = Source/JavaScriptC
 JavaScriptCore.makefile = Makefile.JavaScriptCore
 SUBDIRS += JavaScriptCore
 
-use?(3D_GRAPHICS) {
-    ANGLE.file = Source/ThirdParty/ANGLE/ANGLE.pro
-    ANGLE.makefile = Makefile.ANGLE
-    SUBDIRS += ANGLE
-}
+#use?(3D_GRAPHICS) {
+#    ANGLE.file = Source/ThirdParty/ANGLE/ANGLE.pro
+#    ANGLE.makefile = Makefile.ANGLE
+#    SUBDIRS += ANGLE
+#}
 
 WebCore.file = Source/WebCore/WebCore.pro
 WebCore.makefile = Makefile.WebCore
 
