--- linux/3rdparty/rtl8821ce/Makefile.orig	2019-06-09 22:13:30.699046284 +0300
+++ linux/3rdparty/rtl8821ce/Makefile	2019-06-09 21:43:36.689977150 +0300
@@ -818,7 +818,7 @@ endif
 
 ########### HAL_RTL8821C #################################
 ifeq ($(CONFIG_RTL8821C), y)
-include $(TopDIR)/rtl8821c.mk
+include $(TopDIR)/3rdparty/rtl8821ce/rtl8821c.mk
 endif
 
 ########### AUTO_CFG  #################################
