--- setup.py.orig	2011-02-14 10:50:11.693000025 +0200
+++ setup.py	2011-02-14 10:51:24.851000024 +0200
@@ -1,8 +1,6 @@
 #!/usr/bin/python
 # -*- coding: utf-8 -*-
 #
-# Copyright (C) 2006-2009 TUBITAK/UEKAE
-#
 # This program is free software; you can redistribute it and/or modify it under
 # the terms of the GNU General Public License as published by the Free
 # Software Foundation; either version 2 of the License, or (at your option)
@@ -29,7 +27,7 @@
     os.makedirs(".tmp")
     # Collect UI files
     for filename in glob.glob1("ui", "*.ui"):
-        os.system("/usr/kde/4/bin/pykde4uic -o .tmp/ui_%s.py ui/%s" % (filename.split(".")[0], filename))
+        os.system("/usr/bin/pykde4uic -o .tmp/ui_%s.py ui/%s" % (filename.split(".")[0], filename))
     # Collect Python files
     os.system("cp -R code/* .tmp/")
     # Generate POT file
@@ -58,7 +56,7 @@
         # Copy compiled UIs and RCs
         print "Generating UIs..."
         for filename in glob.glob1("ui", "*.ui"):
-            os.system("/usr/kde/4/bin/pykde4uic -o build/%s/%s.py ui/%s" % (about.modName, filename.split(".")[0], filename))
+            os.system("/usr/bin/pykde4uic -o build/%s/%s.py ui/%s" % (about.modName, filename.split(".")[0], filename))
         #print "Generating RCs..."
         #for filename in glob.glob1("data", "*.qrc"):
         #    os.system("/usr/bin/pyrcc4 data/%s -o build/%s_rc.py" % (filename, filename.split(".")[0]))
@@ -67,9 +65,9 @@
     def run(self):
         os.system("./setup.py build")
         if self.root:
-            kde_dir = "%s/usr/kde/4" % self.root
+            kde_dir = "%s/usr/" % self.root
         else:
-            kde_dir = "/usr/kde/4"
+            kde_dir = "/usr/"
         bin_dir = os.path.join(kde_dir, "bin")
         locale_dir = os.path.join(kde_dir, "share/locale")
         service_dir = os.path.join(kde_dir, "share/kde4/services")
