--- history-manager-0.2.7.4.orig/setup.py	2009-05-27 18:07:26.000000000 +0300
+++ history-manager-0.2.7.4/setup.py	2014-04-18 10:48:38.000000000 +0300
@@ -29,7 +29,7 @@
     os.makedirs(".tmp")
     # Collect UI files
     for filename in glob.glob1("ui", "*.ui"):
-        os.system("/usr/kde/4/bin/pykde4uic -o .tmp/ui_%s.py ui/%s" % (filename.split(".")[0], filename))
+        os.system("/usr/bin/pykde4uic -o .tmp/ui_%s.py ui/%s" % (filename.split(".")[0], filename))
     # Collect Python files
     os.system("cp -R code/* .tmp/")
     # Generate POT file
@@ -58,7 +58,7 @@
         # Copy compiled UIs and RCs
         print "Generating UIs..."
         for filename in glob.glob1("ui", "*.ui"):
-            os.system("/usr/kde/4/bin/pykde4uic -o build/%s/%s.py ui/%s" % (about.modName, filename.split(".")[0], filename))
+            os.system("/usr/bin/pykde4uic -o build/%s/%s.py ui/%s" % (about.modName, filename.split(".")[0], filename))
         #print "Generating RCs..."
         #for filename in glob.glob1("data", "*.qrc"):
         #    os.system("/usr/bin/pyrcc4 data/%s -o build/%s_rc.py" % (filename, filename.split(".")[0]))
@@ -67,14 +67,14 @@
     def run(self):
         os.system("./setup.py build")
         if self.root:
-            kde_dir = "%s/usr/kde/4" % self.root
+            kde_dir = "%s/usr/share/kde4" % self.root
         else:
-            kde_dir = "/usr/kde/4"
+            kde_dir = "/usr/share/kde4"
         bin_dir = os.path.join(kde_dir, "bin")
-        locale_dir = os.path.join(kde_dir, "share/locale")
-        service_dir = os.path.join(kde_dir, "share/kde4/services")
-        apps_dir = os.path.join(kde_dir, "share/applications/kde4")
-        project_dir = os.path.join(kde_dir, "share/apps", about.appName)
+        locale_dir = os.path.join("/usr/share/locale")
+        service_dir = os.path.join(kde_dir, "services")
+        apps_dir = os.path.join(kde_dir, "applications")
+        project_dir = os.path.join(kde_dir, "apps", about.appName)
 
         # Make directories
         print "Making directories..."
