diff --git a/data/new-launcher.nemo_action.in b/data/new-launcher.nemo_action.in
index 58ae01d..c8d4e36 100644
--- a/data/new-launcher.nemo_action.in
+++ b/data/new-launcher.nemo_action.in
@@ -4,7 +4,7 @@ Name=Create a new l_auncher here...
 
 Comment=Create a new launcher in this folder
 
-Exec=cinnamon-desktop-editor -mnemo-launcher -d%P
+Exec=cinnamon-desktop-editor -mnemo-launcher -d"%P"
 
 Selection=None
 
diff --git a/eel/eel-editable-label.c b/eel/eel-editable-label.c
index 496c19a..c4341fa 100644
--- a/eel/eel-editable-label.c
+++ b/eel/eel-editable-label.c
@@ -1550,6 +1550,11 @@ eel_editable_label_draw (GtkWidget *widget,
   label = EEL_EDITABLE_LABEL (widget);
   style = gtk_widget_get_style_context (widget);
 
+  gtk_render_background (style, cr,
+                         0, 0,
+                         gtk_widget_get_allocated_width (widget),
+                         gtk_widget_get_allocated_height (widget));
+
   eel_editable_label_ensure_layout (label, TRUE);
   
   if (gtk_widget_get_visible (widget) && gtk_widget_get_mapped (widget) &&
diff --git a/files/usr/share/nemo/actions/add-desklets.nemo_action b/files/usr/share/nemo/actions/add-desklets.nemo_action
index eb1249e..0aa2884 100644
--- a/files/usr/share/nemo/actions/add-desklets.nemo_action
+++ b/files/usr/share/nemo/actions/add-desklets.nemo_action
@@ -16,82 +16,82 @@ Dependencies=cinnamon-settings;
 
 Conditions=desktop;dbus org.Cinnamon;
 Name[af]=Voeg Desklets By
-Name[vi]=Thêm Tiểu trình màn hình
 Name[is]=Bæta við skjáborðsgræjum (desklets)
+Name[vi]=Thêm Tiểu trình màn hình
+Name[ca]=Afegir Desklets
 Name[am]=ዴስክሌትስ መጨመሪያ
 Name[it]=Aggiungi Desklet
+Name[sv]=Lägg till miniprogram
 Name[cs]=Přidej Desklet
-Name[gd]=Cuir aplaidean-deasc ris
+Name[cy]=Ychwanegu Desgynau
+Name[ar]=أضف مربع عرض على سطح المكتب
 Name[ga]=Cuir le Deiscíní
 Name[eu]=Gehitu deskletak
 Name[et]=Lisa Desklet
-Name[ca]=Afegir Desklets
-Name[en_AU]=Add Desklets
+Name[gl]=Engadir trebellos
 Name[id]=Tambah Desklet
 Name[es]=Añadir Desklets
 Name[ru]=Добавить десклеты
-Name[gl]=Engadir trebellos
 Name[nl]=Desklets toevoegen
 Name[tr]=Masaüstü Uygulamacıkları Ekle
-Name[ja]=デスクレットを追加
+Name[en_AU]=Add Desklets
 Name[ro]=Adaugă desklets
-Name[jv]=Kabeh Desklet
-Name[pl]=Dodaj Desklety
+Name[bn]=ডেস্কলেট যুক্ত করুন
+Name[en_GB]=Add Desklets
 Name[fr]=Ajouter des desklets
-Name[ms]=Tambah Desklet
+Name[uk]=Додати десклети
 Name[pt_BR]=Adicionar Desklets
 Name[hr]=Dodaj desklete
-Name[bn]=ডেস্কলেট যুক্ত করুন
+Name[jv]=Kabeh Desklet
 Name[de]=Desklets hinzufügen
 Name[da]=Tilføj Skrivebordsprogrammer
 Name[bs]=Dodaj Desklet-e
-Name[cy]=Ychwanegu Desgynau
 Name[hu]=Asztali alkalmazás hozzáadása
+Name[ja]=デスクレットを追加
 Name[pt]=Adicionar Mini-Aplicações
 Name[zh_CN]=添加桌面小工具
 Name[sr]=Додај десктоп аплете
-Name[sv]=Lägg till miniprogram
-Name[ar]=أضف مربع عرض على سطح المكتب
-Name[en_GB]=Add Desklets
-Name[uk]=Додати десклети
+Name[gd]=Cuir aplaidean-deasc ris
+Name[pl]=Dodaj Desklety
+Name[ms]=Tambah Desklet
 Name[sl]=Dodaj aplete namizja
 Comment[af]=Voeg Desklets By
-Comment[vi]=Thêm Tiểu trình màn hình
 Comment[is]=Bæta við skjáborðsgræjum (desklets)
+Comment[vi]=Thêm Tiểu trình màn hình
+Comment[ca]=Afegir Desklets
 Comment[am]=ዴስክሌትስ መጨመሪያ
 Comment[it]=Aggiungi Desklet
+Comment[sv]=Lägg till miniprogram
 Comment[cs]=Přidej Desklet
-Comment[gd]=Cuir aplaidean-deasc ris
+Comment[cy]=Ychwanegu Desgynau
+Comment[ar]=أضف مربع عرض على سطح المكتب
 Comment[ga]=Cuir le Deiscíní
 Comment[eu]=Gehitu deskletak
 Comment[et]=Lisa Desklet
-Comment[ca]=Afegir Desklets
-Comment[en_AU]=Add Desklets
+Comment[gl]=Engadir trebellos
 Comment[id]=Tambah Desklet
 Comment[es]=Añadir Desklets
 Comment[ru]=Добавить десклеты
-Comment[gl]=Engadir trebellos
 Comment[nl]=Desklets toevoegen
 Comment[tr]=Masaüstü Uygulamacıkları Ekle
-Comment[ja]=デスクレットを追加
+Comment[en_AU]=Add Desklets
 Comment[ro]=Adaugă desklets
-Comment[jv]=Kabeh Desklet
-Comment[pl]=Dodaj Desklety
+Comment[bn]=ডেস্কলেট যুক্ত করুন
+Comment[en_GB]=Add Desklets
 Comment[fr]=Ajouter des desklets
-Comment[ms]=Tambah Desklet
+Comment[uk]=Додати десклети
 Comment[pt_BR]=Adicionar Desklets
 Comment[hr]=Dodaj desklete
-Comment[bn]=ডেস্কলেট যুক্ত করুন
+Comment[jv]=Kabeh Desklet
 Comment[de]=Desklets hinzufügen
 Comment[da]=Tilføj Skrivebordsprogrammer
 Comment[bs]=Dodaj Desklet-e
-Comment[cy]=Ychwanegu Desgynau
 Comment[hu]=Asztali alkalmazás hozzáadása
+Comment[ja]=デスクレットを追加
 Comment[pt]=Adicionar Mini-Aplicações
 Comment[zh_CN]=添加桌面小工具
 Comment[sr]=Додај десктоп аплете
-Comment[sv]=Lägg till miniprogram
-Comment[ar]=أضف مربع عرض على سطح المكتب
-Comment[en_GB]=Add Desklets
-Comment[uk]=Додати десклети
+Comment[gd]=Cuir aplaidean-deasc ris
+Comment[pl]=Dodaj Desklety
+Comment[ms]=Tambah Desklet
 Comment[sl]=Dodaj aplete namizja
diff --git a/files/usr/share/nemo/actions/change-background.nemo_action b/files/usr/share/nemo/actions/change-background.nemo_action
index 0d961c8..59714b6 100644
--- a/files/usr/share/nemo/actions/change-background.nemo_action
+++ b/files/usr/share/nemo/actions/change-background.nemo_action
@@ -18,97 +18,97 @@ Conditions=desktop;dbus org.Cinnamon;
 
 Name[el]=Αλλαγή _εικόνας επιφάνειας εργασίας
 Name[af]=Verander Werkskerm Agtergrond
-Name[vi]=Đổi Ảnh _nền Màn hình
 Name[is]=Breyta _bakgrunni skjáborðs
+Name[vi]=Đổi Ảnh _nền Màn hình
+Name[ca]=Canvia el _fons de l'escriptori
 Name[am]=የዴስክቶፕ _መደብ መቀየሪያ
 Name[it]=Ca_mbia sfondo scrivania
+Name[sv]=Byt skrivbordsbak_grund
 Name[cs]=Změnit _pozadí plochy
-Name[gd]=Atharraich _cùl-ionad a' bhàrr-deasc
-Name[ml]=ഡെസ്ക്ടോപ്പിന്റെ _പശ്ചാത്തലം മാറ്റുക
+Name[cy]=Newid Cefndir y _Bwrdd Gwaith
+Name[ar]=تغيير _خلفية سطح المكتب
 Name[ga]=Athraigh Cúl_ra na Deisce
 Name[eu]=Aldatu _mahaigainaren atzeko planoa
 Name[et]=Vaheta töölaua _taust
-Name[ca]=Canvia el _fons de l'escriptori
-Name[en_AU]=Change Desktop _Background
+Name[gl]=Cambiar o _fondo do escritorio
 Name[id]=Ubah Latar _Belakang Desktop
 Name[es]=Cambiar el _fondo del escritorio
 Name[ru]=_Изменить фон рабочего стола
-Name[gl]=Cambiar o _fondo do escritorio
 Name[nl]=Werk_bladachtergrond wijzigen
 Name[tr]=Masaüstü _Arkaplanını Değiştir
-Name[ja]=背景を変更 (_B)
+Name[en_AU]=Change Desktop _Background
 Name[lt]=Keisti darbalaukio _foną
 Name[ro]=Schim_bă fundalul desktop-ului
-Name[jv]=Ganti Gambar _Buri Desktop
-Name[pl]=Zmień _tło pulpitu
+Name[bn]=ডেস্কটপের পটভূমি পরিবর্তন করুন (_B)
+Name[en_GB]=Change Desktop _Background
 Name[fr]=Modifier l'arrière-plan du _bureau
 Name[zh_TW]=更改桌面背景(_B)
-Name[ms]=Tukar Latar _Belakang Desktop
+Name[uk]=Змінити т_ло стільниці
 Name[pt_BR]=Alterar plano de fundo
 Name[hr]=Promijeni pozadinu _radne površine
-Name[bn]=ডেস্কটপের পটভূমি পরিবর্তন করুন (_B)
-Name[de]=_Hintergrund der Arbeitsfläche ändern
+Name[jv]=Ganti Gambar _Buri Desktop
+Name[de]=_Hintergrund des Schreibtisches ändern
 Name[sr@latin]=Iz_meni pozadinu
 Name[da]=Skift _skrivebordsbaggrund
 Name[bs]=Promijeni _pozadinu desktopa
-Name[cy]=Newid Cefndir y _Bwrdd Gwaith
 Name[hu]=Asztal _hátterének módosítása
+Name[ja]=背景を変更 (_B)
 Name[fr_CA]=Changer l'_arrière-plan du bureau
+Name[ml]=ഡെസ്ക്ടോപ്പിന്റെ _പശ്ചാത്തലം മാറ്റുക
 Name[pt]=Alterar _fundo da área de trabalho
 Name[zh_CN]=更改桌面背景(_B)
 Name[sr]=Из_мени позадину
-Name[sv]=Byt skrivbordsbak_grund
 Name[sk]=Zmeniť _pozadie plochy
-Name[ar]=تغيير _خلفية سطح المكتب
-Name[en_GB]=Change Desktop _Background
-Name[uk]=Змінити т_ло стільниці
+Name[gd]=Atharraich _cùl-ionad a' bhàrr-deasc
+Name[pl]=Zmień _tło pulpitu
+Name[ms]=Tukar Latar _Belakang Desktop
 Name[sl]=Spremeni _ozadje namizja
 Name[nds]=Desktop_Hintergrund ändern
 Comment[el]=Αλλαγή _εικόνας επιφάνειας εργασίας
 Comment[af]=Verander Werkskerm Agtergrond
-Comment[vi]=Đổi Ảnh _nền Màn hình
 Comment[is]=Breyta _bakgrunni skjáborðs
+Comment[vi]=Đổi Ảnh _nền Màn hình
+Comment[ca]=Canvia el _fons de l'escriptori
 Comment[am]=የዴስክቶፕ _መደብ መቀየሪያ
 Comment[it]=Ca_mbia sfondo scrivania
+Comment[sv]=Byt skrivbordsbak_grund
 Comment[cs]=Změnit _pozadí plochy
-Comment[gd]=Atharraich _cùl-ionad a' bhàrr-deasc
-Comment[ml]=ഡെസ്ക്ടോപ്പിന്റെ _പശ്ചാത്തലം മാറ്റുക
+Comment[cy]=Newid Cefndir y _Bwrdd Gwaith
+Comment[ar]=تغيير _خلفية سطح المكتب
 Comment[ga]=Athraigh Cúl_ra na Deisce
 Comment[eu]=Aldatu _mahaigainaren atzeko planoa
 Comment[et]=Vaheta töölaua _taust
-Comment[ca]=Canvia el _fons de l'escriptori
-Comment[en_AU]=Change Desktop _Background
+Comment[gl]=Cambiar o _fondo do escritorio
 Comment[id]=Ubah Latar _Belakang Desktop
 Comment[es]=Cambiar el _fondo del escritorio
 Comment[ru]=_Изменить фон рабочего стола
-Comment[gl]=Cambiar o _fondo do escritorio
 Comment[nl]=Werk_bladachtergrond wijzigen
 Comment[tr]=Masaüstü _Arkaplanını Değiştir
-Comment[ja]=背景を変更 (_B)
+Comment[en_AU]=Change Desktop _Background
 Comment[lt]=Keisti darbalaukio _foną
 Comment[ro]=Schim_bă fundalul desktop-ului
-Comment[jv]=Ganti Gambar _Buri Desktop
-Comment[pl]=Zmień _tło pulpitu
+Comment[bn]=ডেস্কটপের পটভূমি পরিবর্তন করুন (_B)
+Comment[en_GB]=Change Desktop _Background
 Comment[fr]=Modifier l'arrière-plan du _bureau
 Comment[zh_TW]=更改桌面背景(_B)
-Comment[ms]=Tukar Latar _Belakang Desktop
+Comment[uk]=Змінити т_ло стільниці
 Comment[pt_BR]=Alterar plano de fundo
 Comment[hr]=Promijeni pozadinu _radne površine
-Comment[bn]=ডেস্কটপের পটভূমি পরিবর্তন করুন (_B)
-Comment[de]=_Hintergrund der Arbeitsfläche ändern
+Comment[jv]=Ganti Gambar _Buri Desktop
+Comment[de]=_Hintergrund des Schreibtisches ändern
 Comment[sr@latin]=Iz_meni pozadinu
 Comment[da]=Skift _skrivebordsbaggrund
 Comment[bs]=Promijeni _pozadinu desktopa
-Comment[cy]=Newid Cefndir y _Bwrdd Gwaith
 Comment[hu]=Asztal _hátterének módosítása
+Comment[ja]=背景を変更 (_B)
 Comment[fr_CA]=Changer l'_arrière-plan du bureau
+Comment[ml]=ഡെസ്ക്ടോപ്പിന്റെ _പശ്ചാത്തലം മാറ്റുക
 Comment[pt]=Alterar _fundo da área de trabalho
 Comment[zh_CN]=更改桌面背景(_B)
 Comment[sr]=Из_мени позадину
-Comment[sv]=Byt skrivbordsbak_grund
 Comment[sk]=Zmeniť _pozadie plochy
-Comment[ar]=تغيير _خلفية سطح المكتب
-Comment[en_GB]=Change Desktop _Background
-Comment[uk]=Змінити т_ло стільниці
+Comment[gd]=Atharraich _cùl-ionad a' bhàrr-deasc
+Comment[pl]=Zmień _tło pulpitu
+Comment[ms]=Tukar Latar _Belakang Desktop
 Comment[sl]=Spremeni _ozadje namizja
 Comment[nds]=Desktop_Hintergrund ändern
diff --git a/files/usr/share/nemo/actions/new-launcher.nemo_action b/files/usr/share/nemo/actions/new-launcher.nemo_action
index 5b88910..530d6d3 100644
--- a/files/usr/share/nemo/actions/new-launcher.nemo_action
+++ b/files/usr/share/nemo/actions/new-launcher.nemo_action
@@ -4,7 +4,7 @@ Name=Create a new l_auncher here...
 
 Comment=Create a new launcher in this folder
 
-Exec=cinnamon-desktop-editor -mnemo-launcher -d%P
+Exec=cinnamon-desktop-editor -mnemo-launcher -d"%P"
 
 Selection=None
 
@@ -14,54 +14,59 @@ Extensions=any;
 
 Dependencies=cinnamon-desktop-editor;
 
-Name[vi]=Tạo trình _khởi chạy mới ở đây...
 Name[is]=Búa til nýjan _forritaræsi hér...
+Name[vi]=Tạo trình _khởi chạy mới ở đây...
+Name[ca]=Crear un nou l_lançador aquí ...
 Name[it]=Crea un nuovo l_anciatore qui ...
+Name[sv]=Skapa en ny l_auncher här...
 Name[cs]=Vytvořit zde nový s_pouštěč...
-Name[gd]=Cruthaich lòinsear ùr an-seo...
-Name[ca]=Crear un nou l_lançador aquí ...
+Name[cy]=Creu C_ychwynnydd yma...
+Name[gl]=Crear aquí un novo _iniciador
 Name[id]=Buat sebuah pel_uncur disini...
 Name[es]=Crear un nuevo l_anzador aquí ...
 Name[nl]=Maak hier een nieuwe starter...
 Name[tr]=Burada yeni bir b_aşlatıcı oluştur...
-Name[ja]=新しいランチャをここに作成 (_A) ...
 Name[ro]=Creează un nou l_ansator aici...
-Name[pl]=Utwórz nowy _aktywator tutaj...
+Name[en_GB]=Create a new l_auncher here...
 Name[fr]=Créer un nouveau l_anceur
-Name[ms]=Cipta pe_lancar baru disini...
+Name[uk]=Створити новий значок запуску тут...
 Name[pt_BR]=Criar novo l_ançador aqui...
 Name[hr]=Stvori novi p_okretač ovdje...
 Name[de]=Neuen _Starter hier erstellen …
 Name[da]=Opret en ny o_pstarter her...
+Name[hu]=In_dítóikon létrehozása…
+Name[ja]=新しいランチャをここに作成 (_A) ...
 Name[pt]=Criar um novo L_ançador aqui...
 Name[sr]=Направи нови покретач  овде...
-Name[sv]=Skapa en ny l_auncher här...
-Name[en_GB]=Create a new l_auncher here...
-Name[uk]=Створити новий значок запуску тут...
+Name[gd]=Cruthaich lòinsear ùr an-seo...
+Name[pl]=Utwórz nowy _aktywator tutaj...
+Name[ms]=Cipta pe_lancar baru disini...
 Name[sl]=Ustvari nov z_aganjalnik tukaj ...
-Comment[vi]=Tạo trình khởi chạy mới trong thư mục này
 Comment[is]=Búa til nýjan forritaræsi í þessari möppu
+Comment[vi]=Tạo trình khởi chạy mới trong thư mục này
+Comment[ca]=Crear un nou llançador en aquesta carpeta
 Comment[it]=Crea un nuovo lanciatore in questa cartella
+Comment[sv]=Skapa en ny launcher i denna mapp
 Comment[cs]=Vytvoří nový spouštěč v této složce
-Comment[gd]=Cruthaich lòinsear ùr anns a' phasgan seo
-Comment[ca]=Crear un nou llançador en aquesta carpeta
+Comment[cy]=Creu cychwynydd newydd yn y ffolder
+Comment[gl]=Crear un novo iniciador neste cartafol
 Comment[id]=Buat sebuah peluncur baru didalam folder ini
 Comment[es]=Crear un nuevo lanzador en esta carpeta
 Comment[nl]=Maak een nieuwe starter in deze map
 Comment[tr]=Bu dizin içinde yeni bir başlatıcı oluştur
-Comment[ja]=新しいランチャをこのフォルダに作成
 Comment[ro]=Creează un nou l_ansator în acest dosar
-Comment[pl]=Utwórz nowy aktywator w tym katalogu
+Comment[en_GB]=Create a new launcher in this folder
 Comment[fr]=Créer un nouveau lanceur dans ce dossier
-Comment[ms]=Cipta pelancar baru dalam folder ini
+Comment[uk]=Створити новий значок запуску у цій теці
 Comment[pt_BR]=Criar um novo lançador nesta pasta
 Comment[hr]=Stvori novi pokretač u ovoj mapi
 Comment[de]=Einen neuen Starter in diesem Ordner erstellen
 Comment[da]=Opret en ny opstarter i denne mappe
 Comment[hu]=Új indító létrehozása ebben a mappában
+Comment[ja]=新しいランチャをこのフォルダに作成
 Comment[pt]=Criar um novo Lançador neste directório
 Comment[sr]=Направи нови покретач у овој фасцикли
-Comment[sv]=Skapa en ny launcher i denna mapp
-Comment[en_GB]=Create a new launcher in this folder
-Comment[uk]=Створити новий значок запуску у цій теці
+Comment[gd]=Cruthaich lòinsear ùr anns a' phasgan seo
+Comment[pl]=Utwórz nowy aktywator w tym katalogu
+Comment[ms]=Cipta pelancar baru dalam folder ini
 Comment[sl]=Ustvari nov zaganjalnik v tej mapi
diff --git a/files/usr/share/nemo/actions/set-as-background.nemo_action b/files/usr/share/nemo/actions/set-as-background.nemo_action
index 29de59e..50ce0e1 100644
--- a/files/usr/share/nemo/actions/set-as-background.nemo_action
+++ b/files/usr/share/nemo/actions/set-as-background.nemo_action
@@ -12,78 +12,78 @@ Mimetypes=image/*;
 
 Conditions=dbus org.Cinnamon;
 Name[el]=Ορισμός ως Ταπετσαρίας...
-Name[vi]=Đặt làm Hình nền...
 Name[is]=Gera að veggfóðri...
+Name[vi]=Đặt làm Hình nền...
+Name[ca]=Defineix com a fons de pantalla
 Name[am]=እንደ ግድግዳ ወረቀት ማሰናጃ...
 Name[it]=Imposta come sfondo...
+Name[sv]=Sätt som skrivbordsbakgrund
 Name[cs]=Nastavit jako pozadí plochy...
-Name[gd]=Stèidhich mar pàipear-balla...
-Name[ml]=പശ്ചാത്തലചിത്രമായി സജ്ജീകരിക്കുക...
+Name[cy]=Gosod fel Papur Wal...
+Name[ar]=ضبط كخلفية...
 Name[eu]=Ezarri atzeko plano gisa...
 Name[et]=Sea taustapildiks...
-Name[ca]=Defineix com a fons de pantalla
+Name[gl]=Estabelecer como fondo de pantalla
 Name[id]=Pasang sebagai Wallpaper...
 Name[es]=Establecer como papel tapiz...
 Name[ru]=Установить в качестве фонового рисунка рабочего стола...
-Name[gl]=Estabelecer como fondo de pantalla
 Name[nl]=Gebruik als werkbladachtergrond...
 Name[tr]=Duvarkağıdı olarak ayarla...
-Name[ja]=壁紙に設定...
 Name[ro]=Setați ca imagine de fundal...
-Name[pl]=Ustaw jako tło pulpitu...
+Name[en_GB]=Set as Wallpaper...
 Name[fr]=Définir en tant que fond d'écran...
-Name[ms]=Tetapkan sebagai Kertas Dinding...
+Name[uk]=Встановити як тло...
 Name[pt_BR]=Definir como papel de parede
 Name[hr]=Postavi kao pozadinu...
 Name[de]=Als Hintergrund festlegen …
 Name[sr@latin]=Postavi kao pozadinu...
 Name[da]=Anvend som baggrund
 Name[bs]=Postavi kao pozadinu...
-Name[cy]=Gosod fel Papur Wal...
 Name[hu]=Beállítás háttérképként...
+Name[ja]=壁紙に設定...
+Name[ml]=പശ്ചാത്തലചിത്രമായി സജ്ജീകരിക്കുക...
 Name[pt]=Definir como fundo do Ambiente de Trabalho
 Name[zh_CN]=设置为壁纸...
 Name[sr]=Постави као позадину...
-Name[sv]=Sätt som skrivbordsbakgrund
-Name[ar]=ضبط كخلفية...
-Name[en_GB]=Set as Wallpaper...
-Name[uk]=Встановити як тло...
+Name[gd]=Stèidhich mar pàipear-balla...
+Name[pl]=Ustaw jako tło pulpitu...
+Name[ms]=Tetapkan sebagai Kertas Dinding...
 Name[sl]=Nastavi kot sliko ozadja ...
 Comment[el]=Ορισμός της επιλεγμένης εικόνας ως ταπετσαρία στην επιφάνεια εργασίας σας
-Comment[vi]=Đặt ảnh được chọn làm hình nền màn hình
 Comment[is]=Nota valda mynd sem bakgrunn skjáborðsins
+Comment[vi]=Đặt ảnh được chọn làm hình nền màn hình
+Comment[ca]=Utiltiza la imatge seleccionada com fons de pantalla del teu escriptori
 Comment[am]=የተመረጠውን ምስል እንደ ዴስክቶፕ የግድግዳ ወረቀት ማሰናጃ
 Comment[it]=Imposta l'immagine selezionata come sfondo
+Comment[sv]=Använd den markerade bilden som din skrivbordsbakgrund
 Comment[cs]=Použít vybraný obrázek jako pozadí plochy
-Comment[gd]=Stèidhich an ìomhaigh taghte mar do phàipear-balla a' bhàrr-deasc
-Comment[ml]=തിരഞ്ഞെടുത്ത ചിത്രം താങ്കളുടെ ഡെസ്ക്‌ടോപ്പിന്റെ പശ്ചാത്തലചിത്രമായി സജ്ജീകരിക്കുക
+Comment[cy]=Gosod y ddelwedd fel papur wal eich bwrdd gwaith
+Comment[ar]=حدد الصورة المحددة كخلفية لسطح المكتب
 Comment[eu]=Ezarri hautatutako irudia zure mahaigainaren atzeko plano gisa
 Comment[et]=Määra valitud pildid oma töölaua taustapildiks
-Comment[ca]=Utiltiza la imatge seleccionada com fons de pantalla del teu escriptori
+Comment[gl]=Estabelecer a imaxe seleccionada como fondo de pantalla do seu escritorio
 Comment[id]=Pasang gambar yang dipilih sebagai desktop wallpaper
 Comment[es]=Establecer la imagen seleccionada como tu papael tapiz
 Comment[ru]=Установить выбранное изображение в качестве фонового рисунка рабочего стола
-Comment[gl]=Estabelecer a imaxe seleccionada como fondo de pantalla do seu escritorio
 Comment[nl]=Gebruik het geselecteerde bestand als werkbladachtergrond
 Comment[tr]=Seçilen resmi masaüstü duvarkağıdı olarak ayarla
-Comment[ja]=選択した画像をデスクトップの壁紙に設定
 Comment[ro]=Setați imaginea selectată ca imagine de fundal pe desktop
-Comment[pl]=Ustaw wybrany obraz jako tło pulpitu
+Comment[en_GB]=Set the selected image as your desktop wallpaper
 Comment[fr]=Utiliser l'image sélectionnée en tant que fond d'écran pour votre bureau
-Comment[ms]=Tetapkan imej terpilih sebagai kertas dinding desktop anda
+Comment[uk]=Встановити обраний малюнок, як тло стільниці
 Comment[pt_BR]=Definir a imagem selecionada como seu papel de parede
 Comment[hr]=Postavite označenu sliku kao pozadinu radne površine
 Comment[de]=Das ausgewählte Bild als HIntergrund festlegen
 Comment[sr@latin]=Postavi izabranu sliku kao pozadinu na radnu površ
 Comment[da]=Anvend det markerede billede som din skrivebordsbaggrund
 Comment[bs]=Postavite označenu sliku kao desktop pozadinu
-Comment[cy]=Gosod y ddelwedd fel papur wal eich bwrdd gwaith
 Comment[hu]=A kiválasztott kép beállítása az asztal háttereként
-Comment[pt]=Definir a imagem seleccionada como fundo do Ambiente de Trabalho
+Comment[ja]=選択した画像をデスクトップの壁紙に設定
+Comment[ml]=തിരഞ്ഞെടുത്ത ചിത്രം താങ്കളുടെ ഡെസ്ക്‌ടോപ്പിന്റെ പശ്ചാത്തലചിത്രമായി സജ്ജീകരിക്കുക
+Comment[pt]=Definir imagem seleccionada como fundo da área de trabalho
 Comment[zh_CN]=将所选图像设置为您的桌面壁纸
 Comment[sr]=Постави изабрану слику као позадину на радну површ
-Comment[sv]=Använd den markerade bilden som din skrivbordsbakgrund
-Comment[ar]=حدد الصورة المحددة كخلفية لسطح المكتب
-Comment[en_GB]=Set the selected image as your desktop wallpaper
-Comment[uk]=Встановити обраний малюнок, як тло стільниці
+Comment[gd]=Stèidhich an ìomhaigh taghte mar do phàipear-balla a' bhàrr-deasc
+Comment[pl]=Ustaw wybrany obraz jako tło pulpitu
+Comment[ms]=Tetapkan imej terpilih sebagai kertas dinding desktop anda
 Comment[sl]=Nastavi izbrano sliko kot vašo sliko ozadja namizja
diff --git a/libnemo-private/nemo-pathbar-button.c b/libnemo-private/nemo-pathbar-button.c
index 24375a8..e1757e8 100644
--- a/libnemo-private/nemo-pathbar-button.c
+++ b/libnemo-private/nemo-pathbar-button.c
@@ -262,7 +262,7 @@ void
 nemo_pathbar_button_get_preferred_size (GtkWidget *button, GtkRequisition *requisition, gint height)
 {
     GtkRequisition req;
-    gtk_widget_get_preferred_size (button, &req, NULL);
+    gtk_widget_get_preferred_size (button, NULL, &req);
     gint offset = rintf ((float) req.height / PATHBAR_BUTTON_OFFSET_FACTOR) + 4;
     if (!NEMO_PATHBAR_BUTTON (button)->is_left_end) {
         req.width -= offset;
diff --git a/libnemo-private/org.nemo.gschema.xml.in b/libnemo-private/org.nemo.gschema.xml.in
index aef1da9..36375e3 100644
--- a/libnemo-private/org.nemo.gschema.xml.in
+++ b/libnemo-private/org.nemo.gschema.xml.in
@@ -33,9 +33,10 @@
     <value nick="name" value="1"/>
     <value nick="size" value="2"/>
     <value nick="type" value="3"/>
-    <value nick="mtime" value="4"/>
-    <value nick="atime" value="5"/>
-    <value nick="trash-time" value="6"/>
+    <value nick="detailed_type" value="4"/>
+    <value nick="mtime" value="5"/>
+    <value nick="atime" value="6"/>
+    <value nick="trash-time" value="7"/>
   </enum>
 
   <enum id="org.nemo.ZoomLevel">
diff --git a/src/nemo-file-management-properties.c b/src/nemo-file-management-properties.c
index 1b837e7..2f75b29 100644
--- a/src/nemo-file-management-properties.c
+++ b/src/nemo-file-management-properties.c
@@ -119,6 +119,7 @@ static const char * const sort_order_values[] = {
 	"name",
 	"size",
 	"type",
+    "detailed_type",
 	"mtime",
 	"atime",
 	"trash-time",
diff --git a/src/nemo-file-management-properties.glade b/src/nemo-file-management-properties.glade
index ae2da07..1a35424 100644
--- a/src/nemo-file-management-properties.glade
+++ b/src/nemo-file-management-properties.glade
@@ -2676,6 +2676,9 @@ Select additional information to display in the tooltip:&lt;/i&gt;</property>
         <col id="0" translatable="yes">By Type</col>
       </row>
       <row>
+        <col id="0" translatable="yes">By Detailed Type</col>
+      </row>
+      <row>
         <col id="0" translatable="yes">By Modification Date</col>
       </row>
       <row>
diff --git a/src/nemo-icon-view-ui.xml b/src/nemo-icon-view-ui.xml
index 8271284..3391748 100644
--- a/src/nemo-icon-view-ui.xml
+++ b/src/nemo-icon-view-ui.xml
@@ -14,6 +14,7 @@
 					<menuitem name="Sort by Name" action="Sort by Name"/>
 					<menuitem name="Sort by Size" action="Sort by Size"/>
 					<menuitem name="Sort by Type" action="Sort by Type"/>
+					<menuitem name="Sort by Detailed Type" action="Sort by Detailed Type"/>
 					<menuitem name="Sort by Modification Date" action="Sort by Modification Date"/>
 					<menuitem name="Sort by Trash Time" action="Sort by Trash Time"/>
 				</placeholder>
@@ -36,6 +37,7 @@
 					<menuitem name="Sort by Name" action="Sort by Name"/>
 					<menuitem name="Sort by Size" action="Sort by Size"/>
 					<menuitem name="Sort by Type" action="Sort by Type"/>
+					<menuitem name="Sort by Detailed Type" action="Sort by Detailed Type"/>
 					<menuitem name="Sort by Modification Date" action="Sort by Modification Date"/>
 					<menuitem name="Sort by Trash Time" action="Sort by Trash Time"/>
 				</placeholder>
diff --git a/src/nemo-icon-view.c b/src/nemo-icon-view.c
index e86972f..e5ab1ea 100644
--- a/src/nemo-icon-view.c
+++ b/src/nemo-icon-view.c
@@ -148,6 +148,13 @@ static const SortCriterion sort_criteria[] = {
 		N_("by _Type"),
 		N_("Keep icons sorted by type in rows")
 	},
+    {
+        NEMO_FILE_SORT_BY_DETAILED_TYPE,
+        "detailed_type",
+        "Sort by Detailed Type",
+        N_("by _Detailed Type"),
+        N_("Keep icons sorted by detailed type in rows")
+    },
 	{
 		NEMO_FILE_SORT_BY_MTIME,
 		"modification date",
@@ -1460,6 +1467,10 @@ static const GtkRadioActionEntry arrange_radio_entries[] = {
     N_("By _Type"), NULL,
     N_("Keep icons sorted by type in rows"),
     NEMO_FILE_SORT_BY_TYPE },
+  { "Sort by Detailed Type", NULL,
+    N_("By _Detailed Type"), NULL,
+    N_("Keep icons sorted by detailed type in rows"),
+    NEMO_FILE_SORT_BY_DETAILED_TYPE },
   { "Sort by Modification Date", NULL,
     N_("By Modification _Date"), NULL,
     N_("Keep icons sorted by modification date in rows"),
diff --git a/src/nemo-pathbar.c b/src/nemo-pathbar.c
index cba79e9..830f4f3 100644
--- a/src/nemo-pathbar.c
+++ b/src/nemo-pathbar.c
@@ -540,7 +540,7 @@ nemo_path_bar_size_allocate (GtkWidget     *widget,
         GtkAllocation child_allocation;
         GList *list, *first_button;
         gint width;
-        gint allocation_width;
+        gint largest_width;
         gboolean need_sliders;
         gint up_slider_offset;
         gint down_slider_offset;
@@ -567,7 +567,6 @@ nemo_path_bar_size_allocate (GtkWidget     *widget,
                 return;
 	}
         direction = gtk_widget_get_direction (widget);
-        allocation_width = allocation->width;
 
   	/* First, we check to see if we need the scrollbars. */
   	if (path_bar->fake_root) {
@@ -591,7 +590,7 @@ nemo_path_bar_size_allocate (GtkWidget     *widget,
         }
     }
 
-    if (width <= allocation_width) {
+    if (width <= allocation->width) {
         if (path_bar->fake_root) {
             first_button = path_bar->fake_root;
         } else {
@@ -623,7 +622,7 @@ nemo_path_bar_size_allocate (GtkWidget     *widget,
             child = BUTTON_DATA (list->data)->button;
             nemo_pathbar_button_get_preferred_size (child, &child_requisition, allocation->height);
 
-            if (width + child_requisition.width + path_bar->spacing + slider_space > allocation_width) {
+            if (width + child_requisition.width + path_bar->spacing + slider_space > allocation->width) {
                 reached_end = TRUE;
             } else {
                 if (list == path_bar->fake_root) {
@@ -641,7 +640,7 @@ nemo_path_bar_size_allocate (GtkWidget     *widget,
             child = BUTTON_DATA (first_button->next->data)->button;
             nemo_pathbar_button_get_preferred_size (child, &child_requisition, allocation->height);
 
-            if (width + child_requisition.width + path_bar->spacing + slider_space > allocation_width) {
+            if (width + child_requisition.width + path_bar->spacing + slider_space > allocation->width) {
                 reached_end = TRUE;
             } else {
                 width += child_requisition.width + path_bar->spacing;
@@ -671,6 +670,12 @@ nemo_path_bar_size_allocate (GtkWidget     *widget,
             }
     }
 
+    /* Determine the largest possible allocation size */
+    largest_width = allocation->width;
+    if (need_sliders) {
+        largest_width -= (path_bar->spacing + path_bar->slider_width) * 2;
+    }
+
     gboolean first_element = TRUE;
     for (list = first_button; list; list = list->prev) {
         child = BUTTON_DATA (list->data)->button;
@@ -680,11 +685,11 @@ nemo_path_bar_size_allocate (GtkWidget     *widget,
         else
             gtk_label_set_width_chars (GTK_LABEL (BUTTON_DATA (list->data)->pre_padding), 2);
 
-        gtk_widget_get_preferred_size (child, &child_requisition, NULL);
+        gtk_widget_get_preferred_size (child, NULL, &child_requisition);
 
         gtk_widget_get_allocation (widget, &widget_allocation);
 
-        child_allocation.width = child_requisition.width;
+        child_allocation.width = MIN (child_requisition.width, largest_width);
         if (direction == GTK_TEXT_DIR_RTL) {
             child_allocation.x -= child_allocation.width;
         }
@@ -695,7 +700,7 @@ nemo_path_bar_size_allocate (GtkWidget     *widget,
             }
         } else {
             if (need_sliders && direction == GTK_TEXT_DIR_LTR) {
-                if (child_allocation.x + child_allocation.width + path_bar->spacing + path_bar->slider_width - (offset) > widget_allocation.x + allocation_width) {
+                if (child_allocation.x + child_allocation.width + path_bar->spacing + path_bar->slider_width - (offset) > widget_allocation.x + allocation->width) {
                     break;
                 }
             }
@@ -1742,6 +1747,10 @@ make_directory_button (NemoPathBar  *path_bar,
     }
     gtk_box_pack_start (GTK_BOX (child), get_padding_widget(2), FALSE, FALSE, 0);
 
+	if (button_data->label != NULL) {
+		gtk_label_set_ellipsize (GTK_LABEL (button_data->label), PANGO_ELLIPSIZE_MIDDLE);
+	}
+
 	if (button_data->path == NULL) {
         	button_data->path = g_object_ref (path);
 	}