diff -upr mate-menus-1.6.0-orig/layout/mate-settings.menu mate-menus-1.6.0/layout/mate-settings.menu
--- mate-menus-1.6.0-orig/layout/mate-settings.menu	2013-03-31 23:16:30.000000000 +0200
+++ mate-menus-1.6.0/layout/mate-settings.menu	2013-06-04 14:48:13.015124073 +0200
@@ -33,6 +33,7 @@
         </Not>
       </And>
     </Include>
+    <MergeFile>mate-preferences-categories.menu</MergeFile>
   </Menu>
 
   <!-- System Settings -->