From adbd9c70d4326e2577c4b5eff486f1809777fa90 Mon Sep 17 00:00:00 2001
From: rpmbuild <rpmbuild@fedoraproject.org>
Date: Mon, 26 May 2014 10:01:33 +0200
Subject: [PATCH] disable PSD import test which deadlocks on ARM

---
 filter/Module_filter.mk | 1 -
 1 file changed, 1 deletion(-)

diff --git a/filter/Module_filter.mk b/filter/Module_filter.mk
index 3558f44..3f3301f 100644
--- a/filter/Module_filter.mk
+++ b/filter/Module_filter.mk
@@ -87,7 +87,6 @@ $(eval $(call gb_Module_add_check_targets,filter,\
     CppunitTest_filter_pcx_test \
     CppunitTest_filter_pict_test \
     CppunitTest_filter_ppm_test \
-    CppunitTest_filter_psd_test \
     CppunitTest_filter_ras_test \
     CppunitTest_filter_tiff_test \
     CppunitTest_filter_tga_test \
-- 
1.9.0

