diff -Nuar mdicconv-0.2.6.1.OSmanOS/cmake/modules/FindSqlite.cmake mdicconv-0.2.6.1/cmake/modules/FindSqlite.cmake
--- mdicconv-0.2.6.1.OSmanOS/cmake/modules/FindSqlite.cmake        1970-01-01 02:00:00.000000000 +0200
+++ mdicconv-0.2.6.1/cmake/modules/FindSqlite.cmake        2013-12-11 18:55:35.526285279 +0200
@@ -0,0 +1,50 @@
+# - Try to find Sqlite
+# Once done this will define
+#
+#  SQLITE_FOUND - system has Sqlite
+#  SQLITE_INCLUDE_DIR - the Sqlite include directory
+#  SQLITE_LIBRARIES - Link these to use Sqlite
+#  SQLITE_DEFINITIONS - Compiler switches required for using Sqlite
+# Redistribution and use is allowed according to the terms of the BSD license.
+# For details see the accompanying COPYING-CMAKE-SCRIPTS file.
+#
+
+
+# Copyright (c) 2008, Gilles Caulier, <caulier.gilles@gmail.com>
+#
+# Redistribution and use is allowed according to the terms of the BSD license.
+# For details see the accompanying COPYING-CMAKE-SCRIPTS file.
+
+if ( SQLITE_INCLUDE_DIR AND SQLITE_LIBRARIES )
+   # in cache already
+   SET(Sqlite_FIND_QUIETLY TRUE)
+endif ( SQLITE_INCLUDE_DIR AND SQLITE_LIBRARIES )
+
+# use pkg-config to get the directories and then use these values
+# in the FIND_PATH() and FIND_LIBRARY() calls
+if( NOT WIN32 )
+  find_package(PkgConfig)
+
+  pkg_check_modules(PC_SQLITE QUIET sqlite3)
+
+  set(SQLITE_DEFINITIONS ${PC_SQLITE_CFLAGS_OTHER})
+endif( NOT WIN32 )
+
+find_path(SQLITE_INCLUDE_DIR NAMES sqlite3.h
+  PATHS
+  ${PC_SQLITE_INCLUDEDIR}
+  ${PC_SQLITE_INCLUDE_DIRS}
+)
+
+find_library(SQLITE_LIBRARIES NAMES sqlite3
+  PATHS
+  ${PC_SQLITE_LIBDIR}
+  ${PC_SQLITE_LIBRARY_DIRS}
+)
+
+include(FindPackageHandleStandardArgs)
+find_package_handle_standard_args(Sqlite DEFAULT_MSG SQLITE_INCLUDE_DIR SQLITE_LIBRARIES )
+
+# show the SQLITE_INCLUDE_DIR and SQLITE_LIBRARIES variables only in the advanced view
+mark_as_advanced(SQLITE_INCLUDE_DIR SQLITE_LIBRARIES )
+
diff -Nuar mdicconv-0.2.6.1.OSmanOS/src/mdicbuilder.cpp mdicconv-0.2.6.1/src/mdicbuilder.cpp
--- mdicconv-0.2.6.1.OSmanOS/src/mdicbuilder.cpp        2008-07-29 17:34:44.000000000 +0300
+++ mdicconv-0.2.6.1/src/mdicbuilder.cpp        2013-12-11 18:54:11.052287775 +0200
@@ -9,6 +9,10 @@
 // Copyright: See COPYING file that comes with this distribution
 //
 //
+#include <stdio.h>
+#include <stdlib.h>
+#include <string>
+#include <sqlite3.h>
 #include "mdicbuilder.h"
 using namespace std;