Index: examples/Jamfile
===================================================================
--- examples/Jamfile	(revision 3536)
+++ examples/Jamfile	(working copy)
@@ -18,7 +18,7 @@
 
 SubDir TOP examples ;
 
-ADDONLIB = ;
+ADDONLIB = -lX11 ;
 
 if $(SHARED) {
 	# required by shared libraries
Index: test/Jamfile
===================================================================
--- test/Jamfile	(revision 3536)
+++ test/Jamfile	(working copy)
@@ -21,6 +21,9 @@
 # shared library requires linkage with FLTK
 if $(SHARED) {
 	ADDONLIB = $(FLTK_LIBS_FULL) ;
+	ADDONLIB += -lX11 ;
+} else {
+	ADDONLIB = -lX11 ;
 }
 
 TEST_FILES = 
