Index: f-spot-0.8.2/configure
===================================================================
--- f-spot-0.8.2.orig/configure
+++ f-spot-0.8.2/configure
@@ -16190,7 +16190,7 @@ fi
 ENABLE_RELEASE=$enable_release
 
 if test "x$enable_release" = "xyes"; then
-	GMCS_FLAGS="-define:RELEASE"
+	GMCS_FLAGS="-define:RELEASE -sdk:4"
 # FIXME: Enable as soon as we rock sufficiently
 #else
 #	GMCS_FLAGS="-warnaserror"
Index: f-spot-0.8.2/configure.ac
===================================================================
--- f-spot-0.8.2.orig/configure.ac
+++ f-spot-0.8.2/configure.ac
@@ -268,7 +268,7 @@ AM_CONDITIONAL(ENABLE_RELEASE, test "x$e
 ENABLE_RELEASE=$enable_release
 AC_SUBST(ENABLE_RELEASE)
 if test "x$enable_release" = "xyes"; then
-	GMCS_FLAGS="-define:RELEASE"
+	GMCS_FLAGS="-define:RELEASE -sdk:4"
 # FIXME: Enable as soon as we rock sufficiently
 #else
 #	GMCS_FLAGS="-warnaserror"
