--- a/panels/region/cinnamon-region-panel-lang.c
+++ b/panels/region/cinnamon-region-panel-lang.c
@@ -36,6 +36,46 @@
 static GDBusProxy *proxy = NULL;
 
 static void
+logout_requested (GtkButton *button,
+		  gpointer   user_data)
+{
+	GError     *error;
+	GDBusProxy *sm_proxy;
+	GVariant *res;
+
+	error = NULL;
+	sm_proxy = g_dbus_proxy_new_for_bus_sync (G_BUS_TYPE_SESSION,
+						  G_DBUS_PROXY_FLAGS_NONE,
+						  NULL,
+						  "org.gnome.SessionManager",
+						  "/org/gnome/SessionManager",
+						  "org.gnome.SessionManager",
+						  NULL,
+						  &error);
+
+	if (sm_proxy == NULL) {
+		g_warning ("Failed to contact gnome session service: %s", error->message);
+		g_error_free (error);
+		return;
+	}
+
+	res = g_dbus_proxy_call_sync (sm_proxy,
+				      "Logout",
+				      g_variant_new ("(u)", 0),
+				      G_DBUS_CALL_FLAGS_NONE,
+				      -1,
+				      NULL,
+				      &error);
+	if (res == NULL) {
+		g_warning ("Logout failed: %s", error->message);
+		g_error_free (error);
+	} else
+		g_variant_unref (res);
+
+	g_object_unref (sm_proxy);
+}
+
+static void
 selection_changed (GtkTreeSelection *selection,
                    GtkBuilder       *builder)
 {
@@ -46,6 +86,7 @@ selection_changed (GtkTreeSelection *sel
 	GVariant *variant;
 	GError *error = NULL;
 	char *object_path;
+	GtkWidget *widget;
 
 	if (gtk_tree_selection_get_selected (selection, &model, &iter) == FALSE) {
 		g_warning ("No selected languages, this shouldn't happen");
@@ -114,7 +155,10 @@ selection_changed (GtkTreeSelection *sel
         formats_update_language (builder, locale);
         system_update_language (builder, locale);
 
-	/* And done */
+	/* And done, ask for logout */
+	//FIXME we should only do that if it wasn't the original language we started the session with
+	widget = (GtkWidget *)gtk_builder_get_object (builder, "logout_button");
+	gtk_widget_show (widget);
 
 bail:
 	if (variant != NULL)
@@ -208,7 +252,11 @@ setup_language (GtkBuilder *builder)
 	/* Connect buttons */
 	widget = (GtkWidget *)gtk_builder_get_object (builder, "language_add");
 	g_signal_connect (widget, "clicked",
-			  G_CALLBACK (add_language), treeview);	
+			  G_CALLBACK (add_language), treeview);
+
+	widget = (GtkWidget *)gtk_builder_get_object (builder, "logout_button");
+	g_signal_connect (widget, "clicked",
+			  G_CALLBACK (logout_requested), NULL);	
 
 	/* Setup accounts service */
 	proxy = g_dbus_proxy_new_for_bus_sync (G_BUS_TYPE_SYSTEM,
--- a/panels/region/cinnamon-region-panel.ui
+++ b/panels/region/cinnamon-region-panel.ui
@@ -112,11 +112,35 @@
                     <property name="border_width">12</property>
                     <property name="spacing">12</property>
                     <child>
-                      <object class="GtkLabel" id="label4">
+                      <object class="GtkBox" id="box1">
                         <property name="visible">True</property>
                         <property name="can_focus">False</property>
-                        <property name="xalign">0</property>
-                        <property name="label" translatable="yes">Select a display language (change will be applied next time you log in)</property>
+                        <child>
+                          <object class="GtkLabel" id="label4">
+                            <property name="visible">True</property>
+                            <property name="can_focus">False</property>
+                            <property name="xalign">0</property>
+                            <property name="label" translatable="yes">Select a display language</property>
+                          </object>
+                          <packing>
+                            <property name="expand">True</property>
+                            <property name="fill">True</property>
+                            <property name="position">0</property>
+                          </packing>
+                        </child>
+                        <child>
+                          <object class="GtkButton" id="logout_button">
+                            <property name="label" translatable="yes">Log out for changes to take effect</property>
+                            <property name="visible">False</property>
+                            <property name="can_focus">True</property>
+                            <property name="receives_default">True</property>
+                          </object>
+                          <packing>
+                            <property name="expand">False</property>
+                            <property name="fill">True</property>
+                            <property name="position">1</property>
+                          </packing>
+                        </child>
                       </object>
                       <packing>
                         <property name="expand">False</property>
@@ -153,30 +177,33 @@
                           </packing>
                         </child>
                         <child>
-                          <object class="GtkToolbar" id="language-toolbar">
-                            <property name="visible">True</property>
+                          <object class="GtkVBox" id="vbox3">
                             <property name="can_focus">False</property>
-                            <property name="toolbar_style">icons</property>
-                            <property name="show_arrow">False</property>
-                            <property name="icon_size">1</property>
-                            <style>
-                              <class name="inline-toolbar"/>
-                            </style>
-                            <style>
-                              <class name="inline-toolbar"/>
-                            </style>
                             <child>
-                              <object class="GtkToolButton" id="language_add">
-                                <property name="use_action_appearance">False</property>
+                              <object class="GtkLabel" id="label23">
                                 <property name="visible">True</property>
                                 <property name="label" translatable="yes">Add Language</property>
                                 <property name="can_focus">False</property>
                                 <property name="use_underline">True</property>
-                                <property name="icon_name">list-add-symbolic</property>
+                              </object>
+                              <packing>
+                                <property name="expand">True</property>
+                                <property name="fill">True</property>
+                                <property name="position">0</property>
+                              </packing>
+                            </child>
+                            <child>
+                              <object class="GtkLinkButton" id="linkbutton1">
+                                <property name="label" translatable="yes">Install languages...</property>
+                                <property name="visible">True</property>
+                                <property name="can_focus">True</property>
+                                <property name="receives_default">True</property>
+                                <property name="has_tooltip">True</property>
                               </object>
                               <packing>
                                 <property name="expand">False</property>
-                                <property name="homogeneous">True</property>
+                                <property name="fill">True</property>
+                                <property name="position">1</property>
                               </packing>
                             </child>
                           </object>
@@ -187,33 +214,40 @@
                           </packing>
                         </child>
                         <child>
-                          <object class="GtkVBox" id="vbox3">
+                          <object class="GtkToolbar" id="language-toolbar">
+                            <property name="visible">True</property>
                             <property name="can_focus">False</property>
+                            <property name="toolbar_style">icons</property>
+                            <property name="show_arrow">False</property>
+                            <property name="icon_size">1</property>
+                            <style>
+                              <class name="inline-toolbar"/>
+                            </style>
                             <child>
-                              <object class="GtkLabel" id="label23">
+                              <object class="GtkToolButton" id="language_add">
                                 <property name="visible">True</property>
                                 <property name="can_focus">False</property>
-                                <property name="use_underline">True</property>
                                 <property name="label" translatable="yes">Add Language</property>
+                                <property name="use_underline">True</property>
+                                <property name="icon_name">list-add-symbolic</property>
                               </object>
                               <packing>
-                                <property name="expand">True</property>
-                                <property name="fill">True</property>
-                                <property name="position">0</property>
+                                <property name="expand">False</property>
+                                <property name="homogeneous">True</property>
                               </packing>
                             </child>
                             <child>
-                              <object class="GtkLinkButton" id="linkbutton1">
-                                <property name="label" translatable="yes">Install languages...</property>
+                              <object class="GtkToolButton" id="language_remove">
                                 <property name="visible">True</property>
-                                <property name="can_focus">True</property>
-                                <property name="receives_default">True</property>
-                                <property name="has_tooltip">True</property>
+                                <property name="sensitive">False</property>
+                                <property name="can_focus">False</property>
+                                <property name="label" translatable="yes">Remove Language</property>
+                                <property name="use_underline">True</property>
+                                <property name="icon_name">list-remove-symbolic</property>
                               </object>
                               <packing>
                                 <property name="expand">False</property>
-                                <property name="fill">True</property>
-                                <property name="position">1</property>
+                                <property name="homogeneous">True</property>
                               </packing>
                             </child>
                           </object>