diff --git a/data/budgie-desktop.desktop.in b/data/budgie-desktop.desktop.in
index 48c8c19..746f823 100644
--- a/data/budgie-desktop.desktop.in
+++ ./data/budgie-desktop.desktop.in
@@ -1,7 +1,7 @@
 [Desktop Entry]
 Name=Budgie Desktop
 Comment=This session logs you into the Budgie Desktop
-Exec=@prefix@/bin/budgie-session
-TryExec=@prefix@/bin/budgie-session
+Exec=@prefix@/bin/budgie-session-launcher
+TryExec=@prefix@/bin/budgie-session-launcher
 Icon=
 Type=Application
 
