diff -Naur mdic.orig/mdic.desktop mdic/mdic.desktop
--- mdic.orig/mdic.desktop        2008-11-22 12:38:51.000000000 +0200
+++ mdic/mdic.desktop        2008-11-23 14:26:19.000000000 +0200
@@ -1,9 +1,11 @@
 [Desktop Entry]
 Categories=Dictionary;Utility;
 Comment=Multilingual Dictionary for GNU/Linux
+Comment[tr]=Linux İçin Çok Dilli Sözlük
 Exec=mdic
 GenericName=MDic Dictionary
+GenericName[tr]=MDic Sözlük
 Icon=mdic
 Name=MDic
 Type=Application