
access_ok() dropped first parameter in 5.0

Signed-off-by: Thomas Backlund <tmb@mageia.org>

--- ./3rdparty/rtl8812au/os_dep/rtw_android.c.orig	2019-03-05 22:28:19.689985714 +0200
+++ ./3rdparty/rtl8812au/os_dep/rtw_android.c	2019-03-06 12:30:59.205473677 +0200
@@ -347,7 +347,7 @@ int rtw_android_priv_cmd(struct net_devi
 		goto exit;
 	}
 
-	if (!access_ok(VERIFY_READ, priv_cmd.buf, priv_cmd.total_len)){
+	if (!access_ok(priv_cmd.buf, priv_cmd.total_len)){
 		DBG_871X("%s: failed to access memory\n", __FUNCTION__);
 		ret = -EFAULT;
 		goto exit;
