--- a/chrome/browser/ui/webui/engagement/site_engagement_ui.cc       2016-03-17 14:15:42.428182847 +0200
+++ b/chrome/browser/ui/webui/engagement/site_engagement_ui.cc   2016-03-17 14:17:05.922185083 +0200
@@ -55,7 +55,7 @@
                                        double score) override {
     GURL origin_gurl(origin.get());
     if (!origin_gurl.is_valid() || score < 0 ||
-        score > SiteEngagementScore::kMaxPoints || isnan(score)) {
+        score > SiteEngagementScore::kMaxPoints || std::isnan(score)) {
       return;
     }
 
 
