diff -Nuar gst-editing-services.orig/autogen.sh gst-editing-services/autogen.sh
--- gst-editing-services.orig/autogen.sh	2013-11-02 22:27:47.000000000 +0200
+++ gst-editing-services/autogen.sh	2013-11-02 22:36:07.768536800 +0200
@@ -17,14 +17,6 @@
 package=gst-editing-services
 srcfile=gst-editing-services.doap
 
-# Make sure we have common
-if test ! -f common/gst-autogen.sh;
-then
-  echo "+ Setting up common submodule"
-  git submodule init
-fi
-git submodule update
-
 # source helper functions
 if test ! -f common/gst-autogen.sh;
 then
@@ -34,13 +26,6 @@
 fi
 . common/gst-autogen.sh
 
-# install pre-commit hook for doing clean commits
-if test ! \( -x .git/hooks/pre-commit -a -L .git/hooks/pre-commit \);
-then
-    rm -f .git/hooks/pre-commit
-    ln -s ../../common/hooks/pre-commit.hook .git/hooks/pre-commit
-fi
-
 # GNU gettext automake support doesn't get along with git.
 # https://bugzilla.gnome.org/show_bug.cgi?id=661128
 if test -d po ; then
