--- jdownloader.desktop.OSmanOS	2009-08-29 21:39:07.000000000 +0300
+++ jdownloader.desktop	2013-07-14 14:08:16.500116549 +0300
@@ -1,7 +1,9 @@
 [Desktop Entry]
+Version=1.0
 Name=JDownloader
-Categories=Network;FileTransfer
+Comment=downloader for One-Click-Hosters like Rapidshare.com or Megaupload.com
 Exec=jdownloader
 Icon=jdownloader
 Terminal=false
+Categories=Network;FileTransfer
 Type=Application
