diff -Naur idjc-0.7.9.orig/idjc.desktop_ idjc-0.7.9/idjc.desktop_
--- idjc-0.7.9.orig/idjc.desktop_	2008-06-17 13:43:37.000000000 +0300
+++ idjc-0.7.9/idjc.desktop_	2008-09-27 03:04:15.000000000 +0300
@@ -1,15 +1,17 @@
 [___Desktop Entry]
-Categories=GNOME;Application;Network;
+Categories=GTK;Application;AudioVideo;Audio;
 Comment=Create your live radio show or podcast
+Comment[tr]=Canlı radyo yayını veya podcast hazırlayın
 Comment[en_GB]=Create your live radio show or podcast
 Comment[de]=Erstelle deine eigene Radiosendung oder Podcasts
 Comment[fr]=Créez votre spectacle radio en direct ou votre podcast
 Exec=EXECPATHNAME
 GenericName=Graphical shoutcast/icecast client
+GenericName[tr]=Shoutcast / Icecast İstemcisi
 GenericName[en_GB]=Graphical shoutcast/icecast client
 GenericName[de]=Grafischer Shoutcast/Icecast Client
 GenericName[fr]=Client graphique shoutcast/icecast
-Icon=ICONPATHNAME
+Icon=idjc
 MimeType=
 Name=Internet DJ Console
 Path=
